[ edit ]
The stages of CRISPR immunity for each of the 3 major types of adaptive immunity. (1) Acquisition begins by recognition of invading DNA by Cas1 and Cas2 and cleavage of a protospacer. (2) The protospacer is ligated to the direct repeat adjacent to the leader sequence and (3) single strand extension repairs the CRISPR and duplicates the direct repeat. The crRNA processing and interference stages occur differently in each of the three major CRISPR systems. (4) The primary CRISPR transcript is cleaved by cas genes to produce crRNAs. (5) In type I systems Cas6e/Cas6f cleave at the junction of ssRNA and dsRNA formed by hairpin loops in the direct repeat. Type II systems use a trans-activating (tracr) RNA to form dsRNA, which is cleaved by Cas9 and RNaseIII. Type III systems use a Cas6 homolog that does not require hairpin loops in the direct repeat for cleavage. (6) In type II and type III systems secondary trimming is performed at either the 5' or 3' end to produce mature crRNAs. (7) Mature crRNAs associate with Cas proteins to form interference complexes. (8) In type I and type II systems, interactions between the protein and PAM sequence are required for degradation of invading DNA. Type III systems do not require a PAM for successful degradation and in type III-A systems basepairing occurs between the crRNA and mRNA rather than the DNA, targeted by type III-B systems.
The CRISPR genetic locus provides bacteria with a defense mechanism to protect them from repeated phage infections.
Transcripts of the CRISPR Genetic Locus and Maturation of pre-crRNA
3D structure of the CRISPR-Cas9 interference complex
CRISPR-Cas9 as a molecular tool introduces targeted double strand DNA breaks.
Double-strand DNA breaks introduced by CRISPR-Cas9 allows further genetic manipulation by exploiting endogenous DNA repair mechanisms.
CRISPR-Cas immunity is a natural process of bacteria and archaea. [81] CRISPR-Cas prevents bacteriophage infection, conjugation and natural transformation by degrading foreign nucleic acids that enter the cell. [39]
Spacer acquisition
[ edit ]
When a microbe is invaded by a bacteriophage , the first stage of the immune response is to capture phage DNA and insert it into a CRISPR locus in the form of a spacer. Cas1 and Cas2 are found in both types of CRISPR-Cas immune systems, which indicates that they are involved in spacer acquisition. Mutation studies confirmed this hypothesis, showing that removal of Cas1 or Cas2 stopped spacer acquisition, without affecting CRISPR immune response. [82] [83] [84] [85] [86]
Multiple Cas1 proteins have been characterised and their structures resolved. [87] [88] [89] Cas1 proteins have diverse amino acid sequences. However, their crystal structures are similar and all purified Cas1 proteins are metal-dependent nucleases/ integrases that bind to DNA in a sequence-independent manner. [63] Representative Cas2 proteins have been characterised and possess either (single strand) ssRNA- [90] or (double strand) dsDNA- [91] [92] specific endoribonuclease activity.
In the I-E system of E. coli, Cas1 and Cas2 form a complex where a Cas2 dimer bridges two Cas1 dimers. [93] In this complex, Cas2 performs a non-enzymatic scaffolding role, [93] binding double-stranded fragments of invading DNA, while Cas1 binds the single-stranded flanks of the DNA and catalyses their integration into CRISPR arrays. [94] [95] [96] New spacers are usually added at the beginning of the CRISPR next to the leader sequence creating a chronological record of viral infections. [97] In E. coli a histone like protein called integration host factor ( IHF ), which binds to the leader sequence, is responsible for the accuracy of this integration. [98] IHF also enhances integration efficiency in the type I-F system of Pectobacterium atrosepticum , [99] but in other systems, different host factors may be required [100]
Protospacer adjacent motifs (PAM)
Main article: Protospacer adjacent motif
Bioinformatic analysis of regions of phage genomes that were excised as spacers (termed protospacers) revealed that they were not randomly selected but instead were found adjacent to short (3â€“5 bp) DNA sequences termed protospacer adjacent motifs (PAM). Analysis of CRISPR-Cas systems showed PAMs to be important for type I and type II, but not type III systems during acquisition. [34] [101] [102] [103] [104] [105] In type I and type II systems, protospacers are excised at positions adjacent to a PAM sequence, with the other end of the spacer cut using a ruler mechanism, thus maintaining the regularity of the spacer size in the CRISPR array. [106] [107] The conservation of the PAM sequence differs between CRISPR-Cas systems and appears to be evolutionarily linked to Cas1 and the leader sequence . [105] [108]
New spacers are added to a CRISPR array in a directional manner, [32] occurring preferentially, [58] [101] [102] [109] [110] but not exclusively, adjacent [104] [107] to the leader sequence. Analysis of the type I-E system from E. coli demonstrated that the first direct repeat adjacent to the leader sequence is copied, with the newly acquired spacer inserted between the first and second direct repeats. [85] [106]
The PAM sequence appears to be important during spacer insertion in type I-E systems. That sequence contains a strongly conserved final nucleotide (nt) adjacent to the first nt of the protospacer. This nt becomes the final base in the first direct repeat. [86] [111] [112] This suggests that the spacer acquisition machinery generates single-stranded overhangs in the second-to-last position of the direct repeat and in the PAM during spacer insertion. However, not all CRISPR-Cas systems appear to share this mechanism as PAMs in other organisms do not show the same level of conservation in the final position. [108] It is likely that in those systems, a blunt end is generated at the very end of the direct repeat and the protospacer during acquisition.
Insertion variants
[ edit ]
Analysis of Sulfolobus solfataricus CRISPRs revealed further complexities to the canonical model of spacer insertion, as one of its six CRISPR loci inserted new spacers randomly throughout its CRISPR array, as opposed to inserting closest to the leader sequence. [107]
Multiple CRISPRs contain many spacers to the same phage. The mechanism that causes this phenomenon was discovered in the type I-E system of E. coli. A significant enhancement in spacer acquisition was detected where spacers already target the phage, even mismatches to the protospacer. This 'priming' requires the Cas proteins involved in both acquisition and interference to interact with each other. Newly acquired spacers that result from the priming mechanism are always found on the same strand as the priming spacer. [86] [111] [112] This observation led to the hypothesis that the acquisition machinery slides along the foreign DNA after priming to find a new protospacer. [112]
Biogenesis
[ edit ]
CRISPR-RNA (crRNA), which later guides the Cas nuclease to the target during the interference step, must be generated from the CRISPR sequence. The crRNA is initially transcribed as part of a single long transcript encompassing much of the CRISPR array. [30] This transcript is then cleaved by Cas proteins to form crRNAs. The mechanism to produce crRNAs differs among CRISPR-Cas systems. In type I-E and type I-F systems, the proteins Cas6e and Cas6f respectively, recognise stem-loops [113] [114] [115] created by the pairing of identical repeats that flank the crRNA. [116] These Cas proteins cleave the longer transcript at the edge of the paired region, leaving a single crRNA along with a small remnant of the paired repeat region.
Type III systems also use Cas6, however, their repeats do not produce stem-loops. Cleavage instead occurs by the longer transcript wrapping around the Cas6 to allow cleavage just upstream of the repeat sequence. [117] [118] [119]
Type II systems lack the Cas6 gene and instead utilize RNaseIII for cleavage. Functional type II systems encode an extra small RNA that is complementary to the repeat sequence, known as a trans-activating crRNA (tracrRNA). [47] Transcription of the tracrRNA and the primary CRISPR transcript results in base pairing and the formation of dsRNA at the repeat sequence, which is subsequently targeted by RNaseIII to produce crRNAs. Unlike the other two systems, the crRNA does not contain the full spacer, which is instead truncated at one end. [72]
CrRNAs associate with Cas proteins to form ribonucleotide complexes that recognize foreign nucleic acids. CrRNAs show no preference between the coding and non-coding strands, which is indicative of an RNA-guided DNA-targeting system. [8] [41] [82] [86] [120] [121] [122] The type I-E complex (commonly referred to as Cascade) requires five Cas proteins bound to a single crRNA. [123] [124]
Interference
[ edit ]
During the interference stage in type I systems, the PAM sequence is recognized on the crRNA-complementary strand and is required along with crRNA annealing. In type I systems correct base pairing between the crRNA and the protospacer signals a conformational change in Cascade that recruits Cas3 for DNA degradation.
Type II systems rely on a single multifunctional protein, Cas9 , for the interference step. [72] Cas9 requires both the crRNA and the tracrRNA to function and cleave DNA using its dual HNH and RuvC/RNaseH-like endonuclease domains. Basepairing between the PAM and the phage genome is required in type II systems. However, the PAM is recognized on the same strand as the crRNA (the opposite strand to type I systems).
Type III systems, like type I require six or seven Cas proteins binding to crRNAs. [125] [126] The type III systems analysed from S. solfataricus and P. furiosus both target the mRNA of phages rather than phage DNA genome, [64] [126] which may make these systems uniquely capable of targeting RNA-based phage genomes. [63] Type III systems were also found to target DNA in addition to RNA using a different Cas protein in the complex, Cas10. [127] The DNA cleavage was shown to be transcription dependent. [128]
The mechanism for distinguishing self from foreign DNA during interference is built into the crRNAs and is therefore likely common to all three systems. Throughout the distinctive maturation process of each major type, all crRNAs contain a spacer sequence and some portion of the repeat at one or both ends. It is the partial repeat sequence that prevents the CRISPR-Cas system from targeting the chromosome as base pairing beyond the spacer sequence signals self and prevents DNA cleavage. [129] RNA-guided CRISPR enzymes are classified as type V restriction enzymes .
Evolution
