# CRISPR


| Cascade (CRISPR-associated complex for antiviral defense) | |||||||
|---|---|---|---|---|---|---|---|
![]() |


[[1]](#cite_note-Mulepati_2014-1)[Entrez](/wiki/Entrez)

[947229](https://www.ncbi.nlm.nih.gov/gene?cmd=retrieve&dopt=default&rn=1&list_uids=947229)[PDB](/wiki/Protein_Data_Bank)

[4QYZ](https://www.rcsb.org/structure/4QYZ)[RefSeq (Prot)](/wiki/National_Center_for_Biotechnology_Information)

[NP_417241.1](https://www.ncbi.nlm.nih.gov/protein/NP_417241.1)[UniProt](/wiki/Swiss-Prot)

[P38036](https://www.uniprot.org/uniprot/P38036)| Search for | |
|---|---|
| Structures |
|

[InterPro](https://www.ebi.ac.uk/interpro/protein/P38036)![](//upload.wikimedia.org/wikipedia/commons/thumb/5/5f/Crispr.png/330px-Crispr.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/5/5f/Crispr.png/330px-Crispr.png)


[[2]](#cite_note-Horvath_2010-2)**CRISPR** ([/ˈkrɪspər/](/wiki/Help:IPA/English); acronym of **clustered regularly interspaced short palindromic repeats**) is a family of [DNA](/wiki/DNA) sequences found in the [genomes](/wiki/Genome) of [prokaryotic](/wiki/Prokaryotic) organisms such as [bacteria](/wiki/Bacteria) and [archaea](/wiki/Archaea). [3] Each sequence within an individual prokaryotic CRISPR is derived from a DNA fragment of a

[bacteriophage](/wiki/Bacteriophage)that had previously infected the prokaryote or one of its ancestors.


[[4]](#cite_note-Hille2018-4)These sequences are used to detect and destroy DNA from similar bacteriophages during subsequent infections. Hence these sequences play a key role in the antiviral (i.e. anti-

[[5]](#cite_note-Rath_2015-5)[phage](/wiki/Bacteriophage)) defense system of prokaryotes and provide a form of heritable,


[[4]](#cite_note-Hille2018-4)[acquired immunity](/wiki/Acquired_immunity).


[[3]](#cite_note-pmid25574773-3)

[[6]](#cite_note-Redman_2016-6)

[[7]](#cite_note-pmid17379808-7)CRISPR is found in approximately 50% of sequenced

[[8]](#cite_note-pmid19095942-8)[bacterial genomes](/wiki/Bacterial_genome)and nearly 90% of sequenced archaea.


[[4]](#cite_note-Hille2018-4)[Cas9](/wiki/Cas9) (or "CRISPR-associated protein 9") is an [enzyme](/wiki/Enzyme) that uses CRISPR sequences as a guide to recognize and open up specific strands of DNA that are complementary to the CRISPR sequence. Cas9 enzymes together with CRISPR sequences form the basis of a technology known as [CRISPR-Cas9](/wiki/CRISPR_gene_editing) that can be used to edit genes within living organisms.[[9]](#cite_note-9) [10] This editing process has a wide variety of applications including basic

[biological](/wiki/Biology)research, development of

[biotechnological](/wiki/Biotechnology)products, and treatment of diseases.


[[11]](#cite_note-Proteintech-11)The development of the CRISPR-Cas9 genome editing technique was recognized by the

[[12]](#cite_note-Hsu2014-12)[Nobel Prize in Chemistry](/wiki/Nobel_Prize_in_Chemistry)in 2020 awarded to

[Emmanuelle Charpentier](/wiki/Emmanuelle_Charpentier)and

[Jennifer Doudna](/wiki/Jennifer_Doudna).


[[13]](#cite_note-Nobel_Foundation-13)

[[14]](#cite_note-NYT-20201007-14)## History

[[edit](/w/index.php?title=CRISPR&action=edit§ion=1)]

The CRISPR/Cas system evolved in nature as a means for bacteria to protect themselves from invading viruses and bacteriophages by inserting pieces of their DNA into the host genome. This allowed the adaptive immune system to respond accordingly on a subsequent infection. It was discovered in * Streptococcus pyogenes* and later found across many other species.

### Repeated sequences

[[edit](/w/index.php?title=CRISPR&action=edit§ion=2)]

The discovery of clustered DNA repeats took place independently in three parts of the world. The first description of what would later be called CRISPR is f[Ishino](/wiki/Yoshizumi_Ishino), et al. in 1987. They accidentally cloned part of a CRISPR sequence together with the iap *gene* *(*isozyme conversion of alkaline phosphatase*)* from * Escherichia coli*.


[[15]](#cite_note-Rawat_2021-15)The organization of the repeats surprised them, as clustered repeated sequences are more typically arranged consecutively, without interspersing sequences.

[[16]](#cite_note-Ishino-1987-16)

[[12]](#cite_note-Hsu2014-12)

[[16]](#cite_note-Ishino-1987-16)In 1993, van Solingen et al., published two articles about a cluster of interrupted [direct repeats](/wiki/Direct_repeat) (DR) in that bacterium. The team recognized the diversity of the sequences that intervened in the direct repeats among different strains of *M. tuberculosis* [17] and used this property to design a typing method called

*, which remains in use.*

[spoligotyping](https://en.wiktionary.org/wiki/spoligotyping)

[[18]](#cite_note-Groenen1993-18)

[[19]](#cite_note-Mojica2016-19)[Mojica](/wiki/Francisco_Mojica) studied the function of repeats in the archaeal [genera](/wiki/Genera) * Haloferax* and

*. Mojica's supervisor surmised that the clustered repeats had a role in segregating replicated DNA into daughter cells during cell division, because plasmids and chromosomes with identical repeat arrays could not coexist in*

[Haloarcula](/wiki/Haloarcula)*. They noted the transcription of the interrupted repeats for the first time; CRISPR's first full characterization.*

[Haloferax volcanii](/wiki/Haloferax_volcanii)

[[19]](#cite_note-Mojica2016-19)By 2000, Mojica and his students, after an automated search of published genomes, identified interrupted repeats in 20 species of microbes as belonging to the same family.

[[20]](#cite_note-Mojica2016b-20)Because those sequences were interspaced, Mojica initially called these sequences "short regularly spaced repeats" (SRSR).

[[21]](#cite_note-Mojica_2000-21)In 2001, Mojica and Jansen, who were searching for additional interrupted repeats, proposed the acronym CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) to encompass the numerous acronyms then in use.

[[22]](#cite_note-Isaacson_2021-22)

[[20]](#cite_note-Mojica2016b-20)In 2002, Tang, et al. showed evidence that CRISPR repeat regions in

[[23]](#cite_note-Barrangou_2013-23)*were transcribed into long RNA molecules subsequently processed into unit-length small RNAs, plus some longer forms of 2, 3, or more spacer-repeat units.*

[Archaeoglobus fulgidus](/wiki/Archaeoglobus_fulgidus)

[[24]](#cite_note-pmid12032318-24)

[[25]](#cite_note-pmid25994611-25)In 2005, [Barrangou](/wiki/Rodolphe_Barrangou) discovered that * S thermophilus*, after iterative phage infection challenges, develops increased phage resistance due to the incorporation of additional CRISPR spacer sequences.


[[26]](#cite_note-Romero_2020-26)

[[27]](#cite_note-rbpont-27)### CRISPR-associated systems

[[edit](/w/index.php?title=CRISPR&action=edit§ion=3)]

A major advance came with Jansen's observation that the prokaryote repeat cluster was accompanied by four homologous genes, CRISPR-associated systems, *Cas* 1–4. The Cas proteins showed [helicase](/wiki/Helicase) and [nuclease](/wiki/Nuclease) [motifs](/wiki/Structural_motif), suggesting a role in the dynamic structure of the CRISPR loci. [28] However, CRISPR's function remained enigmatic.

![](//upload.wikimedia.org/wikipedia/commons/thumb/b/b3/SimpleCRISPR.jpg/250px-SimpleCRISPR.jpg)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/b/b3/SimpleCRISPR.jpg/250px-SimpleCRISPR.jpg)

*cas*genes, a leader sequence, and a repeat-spacer array. Repeats are shown as gray boxes and spacers are colored bars. The arrangement of the three components is not always as shown.


[[29]](#cite_note-pmid20056882-29)In addition, several CRISPRs with similar sequences can be present in a single genome, only one of which is associated with

[[30]](#cite_note-pmid20125085-30)*cas*genes.


[[31]](#cite_note-pmid17521438-31)In 2005, three independent research groups showed that some CRISPR spacers are derived from [phage](/wiki/Phage) DNA and [extrachromosomal DNA](/wiki/Extrachromosomal_DNA) such as [plasmids](/wiki/Plasmid).[[32]](#cite_note-pmid15758212-32)[[33]](#cite_note-pmid15791728-33) [34] In effect, the spacers are fragments of DNA gathered from viruses that previously attacked the cell. The source of the spacers was a sign that the CRISPR-

*cas*system could have a role in adaptive immunity in

[bacteria](/wiki/Bacteria).


[[29]](#cite_note-pmid20056882-29)All three studies proposing this idea were initially rejected by high-profile journals, but eventually appeared elsewhere.

[[35]](#cite_note-pmid25963251-35)

[[36]](#cite_note-Lander_2016-36)The first publication [33] proposing a role of CRISPR-Cas in microbial immunity, by Mojica, et al., predicted a role for the RNA transcript of spacers on target recognition in a mechanism that could be analogous to the

[RNA interference](/wiki/RNA_interference)system used by eukaryotic cells. Koonin and colleagues extended this RNA interference hypothesis by proposing mechanisms of action for the different CRISPR-Cas subtypes according to the predicted function of their proteins.


[[37]](#cite_note-Makarova2006-37)Experimental work by several groups revealed the basic mechanisms of CRISPR-Cas immunity. In 2007, the first experimental evidence that CRISPR was part of the adaptive immune system was published.[[7]](#cite_note-pmid17379808-7) [12] A CRISPR region in

*acquired spacers from the DNA of an infecting*

[Streptococcus thermophilus](/wiki/Streptococcus_thermophilus)[bacteriophage](/wiki/Bacteriophage). The researchers manipulated the resistance of

*S. thermophilus*to different types of phages by adding and deleting spacers whose sequence matched those found in the phages.


[[38]](#cite_note-craze-38)In 2008, Brouns and Van der Oost identified a complex of Cas proteins called Cascade, that in

[[39]](#cite_note-Marraffini2015-39)*E. coli*cut the CRISPR RNA precursor within the repeats into mature spacer-containing RNA molecules called

[CRISPR RNA](/wiki/CRISPR_RNA)(crRNA), which remained bound to the protein complex.

Cascade, crRNA and a helicase/nuclease (

[[40]](#cite_note-Brouns_2008-40)[Cas3](/wiki/Cas3)) were required to provide a bacterial host with immunity against infection by a

[DNA virus](/wiki/DNA_virus). By designing an anti-virus CRISPR, they demonstrated that two orientations of the crRNA (sense/antisense) provided immunity, indicating that the crRNA guides were targeting

[dsDNA](/wiki/DsDNA). That year Marraffini and Sontheimer confirmed that a CRISPR sequence of

*targeted DNA and not RNA to prevent*

[S. epidermidis](/wiki/S._epidermidis)[conjugation](/wiki/Bacterial_conjugation). This finding was at odds with the proposed RNA-interference-like mechanism of CRISPR-Cas immunity, although a CRISPR-Cas system that targets foreign RNA was later found in

*.*

[Pyrococcus furiosus](/wiki/Pyrococcus_furiosus)

[[12]](#cite_note-Hsu2014-12)A 2010 study reported that CRISPR-Cas cuts strands of both phage and plasmid DNA in

[[39]](#cite_note-Marraffini2015-39)*S. thermophilus*.


[[41]](#cite_note-Garneau2010-41)In 2012, Jinek et al., fused crRNA and [tracrRNA](/wiki/TracrRNA) into a single-guide RNA, simplifying Cas9 targeting. [42] Šikšnys, et al., reported that Cas9 from

*can target specific DNA by altering crRNA.*

[S. thermophilus](/wiki/Streptococcus_thermophilus)In 2013, Cong, et al., as well as Mali, et al., applied CRISPR-Cas9 to edit human cell cultures.

[[43]](#cite_note-43)

[[44]](#cite_note-44)In 2015, Liang, et al., edited human tripronuclear

[[45]](#cite_note-45)[zygotes](/wiki/Zygotes), achieving successful cleavage in 28 of 54

[embryos](/wiki/Embryos).


[[46]](#cite_note-46)### Cas9

[[edit](/w/index.php?title=CRISPR&action=edit§ion=4)]

![](//upload.wikimedia.org/wikipedia/commons/thumb/f/fa/Cas12a_vs_Cas9_cleavage_position.svg/250px-Cas12a_vs_Cas9_cleavage_position.svg.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/f/fa/Cas12a_vs_Cas9_cleavage_position.svg/250px-Cas12a_vs_Cas9_cleavage_position.svg.png)

[Cas12a](/wiki/Cas12a)and

[Cas9](/wiki/Cas9)with the position of DNA cleavage shown relative to their

[PAM sequences](/wiki/Protospacer_adjacent_motif)in a zoom-in

A simpler CRISPR system from * S pyogenes* uses

[Cas9](/wiki/Cas9), an endonuclease functioning with two small RNAs—crRNA and tracrRNA—to form a four-component complex.


[[47]](#cite_note-Deltcheva2011-47)In 2012,

[[48]](#cite_note-Barrangou2015-48)[Doudna](/wiki/Jennifer_Doudna)and

[Charpentier](/wiki/Emmanuelle_Charpentier)simplified this into a two-component system by fusing the RNAs into a

[single-guide RNA](/wiki/Guide_RNA#Structure), enabling Cas9 to target and cut specific DNA sequences—a breakthrough that earned them the 2020

[Nobel Prize in Chemistry](/wiki/Nobel_Prize_in_Chemistry).

Parallel work showed the

[[49]](#cite_note-pmid22745249-49)*S. thermophilus*Cas9 could be similarly reprogrammed by altering the crRNA sequence.

These developments spurred genome editing efforts, including demonstrations by groups led by

[[19]](#cite_note-Mojica2016-19)[Zhang](/wiki/Feng_Zhang)and

[Church](/wiki/George_M._Church)showing genome editing in human cells using CRISPR-Cas9.


[[12]](#cite_note-Hsu2014-12)

[[50]](#cite_note-Cong2013-50)

[[51]](#cite_note-Mali2013-51)### Cas12a

[[edit](/w/index.php?title=CRISPR&action=edit§ion=5)]

Cas12a, a Class II Type V CRISPR-associated nuclease, was characterized in 2015 and was formerly known as Cpf1. [5] This nuclease is found in the CRISPR-Cpf1 system of bacteria such as

*Francisella novicida*.


[[6]](#cite_note-Redman_2016-6)The initial designation, derived from a TIGRFAMs protein family definition established in 2012, reflected the prevalence of this CRISPR-Cas subtype in the

[[2]](#cite_note-Horvath_2010-2)*Prevotella*and

*Francisella*lineages. Cas12a exhibits several key distinctions from Cas9: it generates staggered cuts in double-stranded DNA, in contrast to the blunt ends produced by Cas9;

it relies on a 'T-rich'

[[11]](#cite_note-Proteintech-11)[protospacer adjacent motif](/wiki/Protospacer_adjacent_motif)(PAM) (typically 5'-TTTV-3', where V is A, C, or G), offering alternative targeting sites compared to the 'G-rich' PAMs (typically 5'-NGG-3') favored by Cas9;

and it requires only a CRISPR RNA (crRNA) for effective targeting, whereas Cas9 necessitates both a crRNA and a

[[13]](#cite_note-Nobel_Foundation-13)*trans*-activating crRNA (tracrRNA).


[[5]](#cite_note-Rath_2015-5)### Cas13a

[[edit](/w/index.php?title=CRISPR&action=edit§ion=6)]

In 2016, the nuclease (formerly known as C2c2) from the bacterium *Leptotrichia shahii* was characterized by researchers in Zhang's group. Cas13 is an RNA-guided RNA endonuclease, which means that it cleaves single-stranded RNA, but not DNA. Cas13 is guided by its crRNA to a ssRNA target and binds and cleaves the target. Similar to Cas12a, Cas13 remains bound to the target and then cleaves other ssRNA molecules non-discriminately. [52] This collateral cleavage property is exploited in the development of various diagnostic technologies.


[[53]](#cite_note-Nucleic_acid_detection_with_CRISPR-53)

[[54]](#cite_note-54)

[[55]](#cite_note-SPRINT:_a_Cas13a-based_platform_for-55)## Locus structure

[[edit](/w/index.php?title=CRISPR&action=edit§ion=7)]

### Repeats and spacers

[[edit](/w/index.php?title=CRISPR&action=edit§ion=8)]

The CRISPR array is made up of an AT-rich leader sequence followed by short repeats that are separated by unique spacers. [56] CRISPR repeats typically range in size from 28 to 37

[base pairs](/wiki/Base_pair)(bps), though there can be as few as 23 bp and as many as 55 bp.

Some show

[[57]](#cite_note-Barrangou2014-57)[dyad symmetry](/wiki/Dyad_symmetry), implying the formation of a

[secondary structure](/wiki/Nucleic_acid_secondary_structure)such as a

[stem-loop](/wiki/Stem-loop)('hairpin') in the RNA, while others are designed to be unstructured. The size of spacers in different CRISPR arrays is typically 32 to 38 bp (range 21 to 72 bp).

New spacers can appear rapidly as part of the immune response to phage infection.

[[57]](#cite_note-Barrangou2014-57)There are usually fewer than 50 units of the repeat-spacer sequence in a CRISPR array.

[[58]](#cite_note-pmid17894817-58)

[[57]](#cite_note-Barrangou2014-57)### CRISPR RNA structures

[[edit](/w/index.php?title=CRISPR&action=edit§ion=9)]

### Cas genes and CRISPR subtypes

[[edit](/w/index.php?title=CRISPR&action=edit§ion=10)]

Small clusters of *cas* genes are often located next to CRISPR repeat-spacer arrays. Collectively the 93 *cas* genes are grouped into 35 families based on sequence similarity of the encoded proteins. 11 of the 35 families form the *cas* core, which includes the protein families Cas1 through Cas9. A complete CRISPR-Cas locus has at least one gene belonging to the *cas* core.[[59]](#cite_note-Makarova2018-59)

CRISPR-Cas systems fall into two classes. Class 1 systems use a complex of multiple Cas proteins to degrade foreign nucleic acids. Class 2 systems use a single large Cas protein for the same purpose. Class 1 is divided into types I, III, and IV; class 2 is divided into types II, V, and VI. [60] The 6 system types are divided into 33 subtypes.

Each type and most subtypes are characterized by a "signature gene" found almost exclusively in the category. Classification is also based on the complement of

[[61]](#cite_note-Makarova2020-61)*cas*genes that are present. Most CRISPR-Cas systems have a Cas1 protein. The

[phylogeny](/wiki/Phylogeny)of Cas1 proteins generally agrees with the classification system,

but exceptions exist due to module shuffling.

[[62]](#cite_note-Makarova2015-62)Many organisms contain multiple CRISPR-Cas systems suggesting that they are compatible and may share components.

[[59]](#cite_note-Makarova2018-59)

[[63]](#cite_note-pmid22337052-63)The sporadic distribution of the CRISPR-Cas subtypes suggests that the CRISPR-Cas system is subject to

[[64]](#cite_note-pmid23320564-64)[horizontal gene transfer](/wiki/Horizontal_gene_transfer)during microbial

[evolution](/wiki/Evolution).

This table is missing information about UniProt and InterPro cross-reference. (October 2020) |

| Class | Cas type | Cas subtype | Signature protein | Function | Ref. |
|---|---|---|---|---|---|
| 1 | I | — |
|

[[65]](#cite_note-pmid21343909-65)[[66]](#cite_note-ReferenceB-66)[PAM](/wiki/Protospacer_adjacent_motif)sequence. Cas5 is required for processing and stability of crRNAs.[[62]](#cite_note-Makarova2015-62)[[67]](#cite_note-67)[[68]](#cite_note-pmid24728998-68)[[69]](#cite_note-pmid21756346-69)[CRISPR-associated transposons](/wiki/CRISPR-associated_transposons)[[62]](#cite_note-Makarova2015-62)[[Note 1]](#cite_note-<span_class="nowrap">I-U</span>-70)[[70]](#cite_note-Makarova2019-71)[Homolog](/wiki/Homologous_series)of Cas10d and Cse1. Binds CRISPR target RNA and promotes stability of the interference complex[[69]](#cite_note-pmid21756346-69)[[71]](#cite_note-pmid30840895-72)[[62]](#cite_note-Makarova2015-62)[[62]](#cite_note-Makarova2015-62)[[62]](#cite_note-Makarova2015-62)[[71]](#cite_note-pmid30840895-72)[[62]](#cite_note-Makarova2015-62)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[Cas9](/wiki/Cas9)[Nucleases](/wiki/Nuclease)RuvC and HNH together produce[DSBs](/wiki/Double-strand_breaks), and separately can produce single-strand breaks. Ensures the acquisition of functional spacers during adaptation.[[72]](#cite_note-pmid22949671-73)[[73]](#cite_note-74)[[74]](#cite_note-75)[Cas4](/wiki/Cas4)[[75]](#cite_note-76)[[76]](#cite_note-pmid23563642-77)[[60]](#cite_note-Wright2016-60)[[77]](#cite_note-78)[Cas12a](/wiki/Cas12a)(Cpf1)[[70]](#cite_note-Makarova2019-71)[[78]](#cite_note-79)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[Note 2]](#cite_note-<span_class="nowrap">V-U</span>5-80)[CRISPR-associated transposons](/wiki/CRISPR-associated_transposons).[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[60]](#cite_note-Wright2016-60)[[79]](#cite_note-81)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[70]](#cite_note-Makarova2019-71)[[80]](#cite_note-Xu-2021-82)[[80]](#cite_note-Xu-2021-82)## Mechanism

[[edit](/w/index.php?title=CRISPR&action=edit§ion=11)]

![](//upload.wikimedia.org/wikipedia/commons/thumb/1/1c/The_Stages_of_CRISPR_immunity.svg/250px-The_Stages_of_CRISPR_immunity.svg.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/The_Stages_of_CRISPR_immunity.svg/250px-The_Stages_of_CRISPR_immunity.svg.png)

(1) Acquisition begins by recognition of invading DNA by

[Cas1](/wiki/Cas1)and Cas2 and cleavage of a protospacer.

(2) The protospacer is ligated to the direct repeat adjacent to the leader sequence and

(3) single strand extension repairs the CRISPR and duplicates the direct repeat. The crRNA processing and interference stages occur differently in each of the three major CRISPR systems.

(4) The primary CRISPR transcript is cleaved by cas genes to produce crRNAs.

(5) In type I systems Cas6e/Cas6f cleave at the junction of ssRNA and dsRNA formed by hairpin loops in the direct repeat. Type II systems use a trans-activating (tracr) RNA to form dsRNA, which is cleaved by

[Cas9](/wiki/Cas9)and RNaseIII. Type III systems use a Cas6 homolog that does not require hairpin loops in the direct repeat for cleavage.

(6) In type II and type III systems secondary trimming is performed at either the 5' or 3' end to produce mature crRNAs.

(7) Mature crRNAs associate with Cas proteins to form interference complexes.

(8) In type I and type II systems, interactions between the protein and PAM sequence are required for degradation of invading DNA. Type III systems do not require a PAM for successful degradation and in type III-A systems basepairing occurs between the crRNA and mRNA rather than the DNA, targeted by type III-B systems.

![](//upload.wikimedia.org/wikipedia/commons/thumb/0/08/12_Hegasy_Cas9_Immun_Wiki_E_CCBYSA.png/250px-12_Hegasy_Cas9_Immun_Wiki_E_CCBYSA.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/0/08/12_Hegasy_Cas9_Immun_Wiki_E_CCBYSA.png/250px-12_Hegasy_Cas9_Immun_Wiki_E_CCBYSA.png)

![](//upload.wikimedia.org/wikipedia/commons/thumb/1/1c/13_Hegasy_CRISPR_pre_crRNA_Wiki_E_CCBYSA.png/250px-13_Hegasy_CRISPR_pre_crRNA_Wiki_E_CCBYSA.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/13_Hegasy_CRISPR_pre_crRNA_Wiki_E_CCBYSA.png/250px-13_Hegasy_CRISPR_pre_crRNA_Wiki_E_CCBYSA.png)

![](//upload.wikimedia.org/wikipedia/commons/thumb/a/a2/14_Hegasy_Cas9_3D_Complex_Wiki_E_CCBYSA.png/250px-14_Hegasy_Cas9_3D_Complex_Wiki_E_CCBYSA.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/a/a2/14_Hegasy_Cas9_3D_Complex_Wiki_E_CCBYSA.png/250px-14_Hegasy_Cas9_3D_Complex_Wiki_E_CCBYSA.png)

![](//upload.wikimedia.org/wikipedia/commons/thumb/f/f4/15_Hegasy_Cas9_DNA_Tool_Wiki_E_CCBYSA.png/250px-15_Hegasy_Cas9_DNA_Tool_Wiki_E_CCBYSA.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/f/f4/15_Hegasy_Cas9_DNA_Tool_Wiki_E_CCBYSA.png/250px-15_Hegasy_Cas9_DNA_Tool_Wiki_E_CCBYSA.png)

![](//upload.wikimedia.org/wikipedia/commons/thumb/c/c9/16_Hegasy_DNA_Rep_Wiki_E_CCBYSA.png/250px-16_Hegasy_DNA_Rep_Wiki_E_CCBYSA.png)


![](http://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/16_Hegasy_DNA_Rep_Wiki_E_CCBYSA.png/250px-16_Hegasy_DNA_Rep_Wiki_E_CCBYSA.png)

CRISPR-Cas immunity is a natural process of bacteria and archaea. [81] CRISPR-Cas prevents bacteriophage infection,

[conjugation](/wiki/Bacterial_conjugation)and

[natural transformation](/wiki/Transformation_(genetics)#Natural_transformation)by degrading foreign nucleic acids that enter the cell.


[[39]](#cite_note-Marraffini2015-39)### Spacer acquisition

[[edit](/w/index.php?title=CRISPR&action=edit§ion=12)]

When a [microbe](/wiki/Microbe) is invaded by a [bacteriophage](/wiki/Bacteriophage), the first stage of the immune response is to capture phage DNA and insert it into a CRISPR locus in the form of a spacer. [Cas1](/wiki/Cas1) and [Cas2](/wiki/Cas2) are found in both types of CRISPR-Cas immune systems, which indicates that they are involved in spacer acquisition. Mutation studies confirmed this hypothesis, showing that removal of Cas1 or Cas2 stopped spacer acquisition, without affecting CRISPR immune response.[[82]](#cite_note-pmid19120484-84)[[83]](#cite_note-pmid23696746-85)[[84]](#cite_note-pmid22160698-86)[[85]](#cite_note-pmid22402487-87)[[86]](#cite_note-pmid22558257-88)

Multiple Cas1 proteins have been characterised and their structures resolved.[[87]](#cite_note-pmid21219465-89)[[88]](#cite_note-pmid19427858-90) [89] Cas1 proteins have diverse

[amino acid](/wiki/Amino_acid)sequences. However, their crystal structures are similar and all purified Cas1 proteins are metal-dependent nucleases/

[integrases](/wiki/Integrases)that bind to DNA in a sequence-independent manner.

Representative Cas2 proteins have been characterised and possess either (single strand) ssRNA-

[[63]](#cite_note-pmid22337052-63)or (double strand) dsDNA-

[[90]](#cite_note-pmid18482976-92)

[[91]](#cite_note-pmid21139194-93)specific

[[92]](#cite_note-pmid22942283-94)[endoribonuclease](/wiki/Endoribonuclease)activity.

In the I-E system of *E. coli*, Cas1 and Cas2 form a complex where a Cas2 dimer bridges two Cas1 dimers. [93] In this complex, Cas2 performs a non-enzymatic scaffolding role,

binding double-stranded fragments of invading DNA, while Cas1 binds the single-stranded flanks of the DNA and catalyses their integration into CRISPR arrays.

[[93]](#cite_note-pmid24793649-95)

[[94]](#cite_note-pmid25707795-96)

[[95]](#cite_note-pmid26478180-97)New spacers are usually added at the beginning of the CRISPR next to the leader sequence creating a chronological record of viral infections.

[[96]](#cite_note-pmid26503043-98)In

[[97]](#cite_note-99)*E. coli*a

[histone like protein](/wiki/Bacterial_DNA_binding_protein)called integration host factor (

[IHF](/wiki/Bacterial_DNA_binding_protein)), which binds to the leader sequence, is responsible for the accuracy of this integration.

IHF also enhances integration efficiency in the type I-F system of

[[98]](#cite_note-100)

[Pectobacterium atrosepticum](/wiki/Pectobacterium_atrosepticum),but in other systems, different host factors may be required

[[99]](#cite_note-101)

[[100]](#cite_note-102)#### Protospacer adjacent motifs (PAM)

[[edit](/w/index.php?title=CRISPR&action=edit§ion=13)]

Bioinformatic analysis of regions of phage genomes that were excised as spacers (termed protospacers) revealed that they were not randomly selected but instead were found adjacent to short (3–5 bp) DNA sequences termed [protospacer adjacent motifs](/wiki/Protospacer_adjacent_motif) (PAM). Analysis of CRISPR-Cas systems showed PAMs to be important for type I and type II, but not type III systems during acquisition.[[34]](#cite_note-pmid16079334-34)[[101]](#cite_note-pmid18065539-103)[[102]](#cite_note-pmid18065545-104)[[103]](#cite_note-pmid19246744-105)[[104]](#cite_note-pmid19239620-106) [105] In type I and type II systems, protospacers are excised at positions adjacent to a PAM sequence, with the other end of the spacer cut using a ruler mechanism, thus maintaining the regularity of the spacer size in the CRISPR array.


[[106]](#cite_note-pmid23445770-108)The conservation of the PAM sequence differs between CRISPR-Cas systems and appears to be evolutionarily linked to Cas1 and the

[[107]](#cite_note-pmid22834906-109)[leader sequence](/wiki/Leader_sequence_(mRNA)).


[[105]](#cite_note-pmid19143596-107)

[[108]](#cite_note-pmid23403393-110)New spacers are added to a CRISPR array in a directional manner, [32] occurring preferentially,


[[58]](#cite_note-pmid17894817-58)

[[101]](#cite_note-pmid18065539-103)

[[102]](#cite_note-pmid18065545-104)

[[109]](#cite_note-pmid18497291-111)but not exclusively, adjacent

[[110]](#cite_note-pmid21149389-112)

[[104]](#cite_note-pmid19239620-106)to the leader sequence. Analysis of the type I-E system from

[[107]](#cite_note-pmid22834906-109)*E. coli*demonstrated that the first direct repeat adjacent to the leader sequence is copied, with the newly acquired spacer inserted between the first and second direct repeats.


[[85]](#cite_note-pmid22402487-87)

[[106]](#cite_note-pmid23445770-108)The PAM sequence appears to be important during spacer insertion in type I-E systems. That sequence contains a strongly conserved final nucleotide (nt) adjacent to the first nt of the protospacer. This nt becomes the final base in the first direct repeat.[[86]](#cite_note-pmid22558257-88)[[111]](#cite_note-pmid22771574-113) [112] This suggests that the spacer acquisition machinery generates single-stranded overhangs in the second-to-last position of the direct repeat and in the PAM during spacer insertion. However, not all CRISPR-Cas systems appear to share this mechanism as PAMs in other organisms do not show the same level of conservation in the final position.

It is likely that in those systems, a blunt end is generated at the very end of the direct repeat and the protospacer during acquisition.

[[108]](#cite_note-pmid23403393-110)#### Insertion variants

[[edit](/w/index.php?title=CRISPR&action=edit§ion=14)]

Analysis of * Sulfolobus solfataricus* CRISPRs revealed further complexities to the canonical model of spacer insertion, as one of its six CRISPR loci inserted new spacers randomly throughout its CRISPR array, as opposed to inserting closest to the leader sequence.


[[107]](#cite_note-pmid22834906-109)Multiple CRISPRs contain many spacers to the same phage. The mechanism that causes this phenomenon was discovered in the type I-E system of *E. coli*. A significant enhancement in spacer acquisition was detected where spacers already target the phage, even mismatches to the protospacer. This 'priming' requires the Cas proteins involved in both acquisition and interference to interact with each other. Newly acquired spacers that result from the priming mechanism are always found on the same strand as the priming spacer.[[86]](#cite_note-pmid22558257-88)[[111]](#cite_note-pmid22771574-113) [112] This observation led to the hypothesis that the acquisition machinery slides along the foreign DNA after priming to find a new protospacer.


[[112]](#cite_note-pmid22781758-114)### Biogenesis

[[edit](/w/index.php?title=CRISPR&action=edit§ion=15)]

CRISPR-RNA (crRNA), which later guides the Cas nuclease to the target during the interference step, must be generated from the CRISPR sequence. The crRNA is initially transcribed as part of a single long transcript encompassing much of the CRISPR array. [30] This transcript is then cleaved by Cas proteins to form crRNAs. The mechanism to produce crRNAs differs among CRISPR-Cas systems. In type I-E and type I-F systems, the proteins Cas6e and Cas6f respectively, recognise stem-loops


[[113]](#cite_note-pmid21572444-115)

[[114]](#cite_note-pmid21572442-116)created by the pairing of identical repeats that flank the crRNA.

[[115]](#cite_note-pmid20829488-117)These Cas proteins cleave the longer transcript at the edge of the paired region, leaving a single crRNA along with a small remnant of the paired repeat region.

[[116]](#cite_note-pmid17442114-118)Type III systems also use Cas6, however, their repeats do not produce stem-loops. Cleavage instead occurs by the longer transcript wrapping around the Cas6 to allow cleavage just upstream of the repeat sequence.[[117]](#cite_note-pmid19141480-119)[[118]](#cite_note-pmid21300293-120)[[119]](#cite_note-pmid24150936-121)

Type II systems lack the Cas6 gene and instead utilize RNaseIII for cleavage. Functional type II systems encode an extra small RNA that is complementary to the repeat sequence, known as a [trans-activating crRNA](/wiki/Trans-activating_crRNA) (tracrRNA). [47] Transcription of the tracrRNA and the primary CRISPR transcript results in base pairing and the formation of dsRNA at the repeat sequence, which is subsequently targeted by RNaseIII to produce crRNAs. Unlike the other two systems, the crRNA does not contain the full spacer, which is instead truncated at one end.


[[72]](#cite_note-pmid22949671-73)CrRNAs associate with Cas proteins to form ribonucleotide complexes that recognize foreign nucleic acids. CrRNAs show no preference between the coding and non-coding strands, which is indicative of an RNA-guided DNA-targeting system.[[8]](#cite_note-pmid19095942-8)[[41]](#cite_note-Garneau2010-41)[[82]](#cite_note-pmid19120484-84)[[86]](#cite_note-pmid22558257-88)[[120]](#cite_note-pmid21646539-122)[[121]](#cite_note-pmid21166892-123) [122] The type I-E complex (commonly referred to as Cascade) requires five Cas proteins bound to a single crRNA.


[[123]](#cite_note-pmid21460843-125)

[[124]](#cite_note-pmid21938068-126)### Interference

[[edit](/w/index.php?title=CRISPR&action=edit§ion=16)]

During the interference stage in type I systems, the PAM sequence is recognized on the crRNA-complementary strand and is required along with crRNA annealing. In type I systems correct base pairing between the crRNA and the protospacer signals a conformational change in Cascade that recruits [Cas3](/wiki/Cas3) for DNA degradation.

Type II systems rely on a single multifunctional protein, [Cas9](/wiki/Cas9), for the interference step. [72] Cas9 requires both the crRNA and the tracrRNA to function and cleave DNA using its dual HNH and RuvC/RNaseH-like endonuclease domains. Basepairing between the PAM and the phage genome is required in type II systems. However, the PAM is recognized on the same strand as the crRNA (the opposite strand to type I systems).

Type III systems, like type I require six or seven Cas proteins binding to crRNAs.[[125]](#cite_note-pmid22227115-127) [126] The type III systems analysed from

*S. solfataricus*and

*P. furiosus*both target the mRNA of phages rather than phage DNA genome,


[[64]](#cite_note-pmid23320564-64)which may make these systems uniquely capable of targeting RNA-based phage genomes.

[[126]](#cite_note-pmid19945378-128)Type III systems were also found to target DNA in addition to RNA using a different Cas protein in the complex, Cas10.

[[63]](#cite_note-pmid22337052-63)The DNA cleavage was shown to be transcription dependent.

[[127]](#cite_note-Estrella_2016-129)

[[128]](#cite_note-Samai_2015-130)The mechanism for distinguishing self from foreign DNA during interference is built into the crRNAs and is therefore likely common to all three systems. Throughout the distinctive maturation process of each major type, all crRNAs contain a spacer sequence and some portion of the repeat at one or both ends. It is the partial repeat sequence that prevents the CRISPR-Cas system from targeting the chromosome as base pairing beyond the spacer sequence signals self and prevents DNA cleavage. [129] RNA-guided CRISPR enzymes are classified as

[type V restriction enzymes](/wiki/Restriction_enzyme#Type_V).

## Evolution

[[edit](/w/index.php?title=CRISPR&action=edit§ion=17)]

| CRISPR associated protein Cas2 (adaptation RNase) | |||||||||
|---|---|---|---|---|---|---|---|---|---|
![]() |

[Pfam](/wiki/Pfam)

[PF09827](https://www.ebi.ac.uk/interpro/entry/pfam/PF09827)[InterPro](/wiki/InterPro)

[IPR019199](https://www.ebi.ac.uk/interpro/entry/IPR019199)[CDD](/wiki/Conserved_Domain_Database)

[cd09638](https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=cd09638)| Available protein structures: | |
|---|---|
|

[structures](http://pfam.xfam.org/family/PF09827?tab=pdbBlock)/[ECOD](http://prodata.swmed.edu/ecod/complete/search?kw=PF09827)[PDB](/wiki/Protein_Data_Bank)

[RCSB PDB](https://www.rcsb.org/search?q=rcsb_polymer_entity_annotation.annotation_id:PF09827%20AND%20rcsb_polymer_entity_annotation.type:Pfam);[PDBe](https://www.ebi.ac.uk/pdbe/entry/search/index?pfam_accession:PF09827);[PDBj](https://pdbj.org/search/pdb?other_db_select=PFam&other_db_field=PF09827)[PDBsum](/wiki/PDBsum)

[structure summary](https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/GetPfamStr.pl?pfam_id=PF09827)| CRISPR-associated protein CasA/Cse1 (Type I effector DNase) | |||||||||
|---|---|---|---|---|---|---|---|---|---|
| Identifiers | |||||||||
| Symbol | CRISPR_Cse1 | ||||||||
|

[PF09481](https://www.ebi.ac.uk/interpro/entry/pfam/PF09481)[InterPro](/wiki/InterPro)

[IPR013381](https://www.ebi.ac.uk/interpro/entry/IPR013381)[CDD](/wiki/Conserved_Domain_Database)

[cd09729](https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=cd09729)| Available protein structures: | |
|---|---|
|

[structures](http://pfam.xfam.org/family/PF09481?tab=pdbBlock)/[ECOD](http://prodata.swmed.edu/ecod/complete/search?kw=PF09481)[PDB](/wiki/Protein_Data_Bank)

[RCSB PDB](https://www.rcsb.org/search?q=rcsb_polymer_entity_annotation.annotation_id:PF09481%20AND%20rcsb_polymer_entity_annotation.type:Pfam);[PDBe](https://www.ebi.ac.uk/pdbe/entry/search/index?pfam_accession:PF09481);[PDBj](https://pdbj.org/search/pdb?other_db_select=PFam&other_db_field=PF09481)[PDBsum](/wiki/PDBsum)

[structure summary](https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/GetPfamStr.pl?pfam_id=PF09481)| CRISPR associated protein CasC/Cse3/Cas6 (Type I effector RNase) | |||||||||
|---|---|---|---|---|---|---|---|---|---|
![]() |

[Pfam](/wiki/Pfam)

[PF08798](https://www.ebi.ac.uk/interpro/entry/pfam/PF08798)[Pfam](/wiki/Pfam)clan

[CL0362](https://www.ebi.ac.uk/interpro/set/pfam/CL0362/)[InterPro](/wiki/InterPro)

[IPR010179](https://www.ebi.ac.uk/interpro/entry/IPR010179)[CDD](/wiki/Conserved_Domain_Database)

[cd09727](https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=cd09727)| Available protein structures: | |
|---|---|
|

[structures](http://pfam.xfam.org/family/PF08798?tab=pdbBlock)/[ECOD](http://prodata.swmed.edu/ecod/complete/search?kw=PF08798)[PDB](/wiki/Protein_Data_Bank)

[RCSB PDB](https://www.rcsb.org/search?q=rcsb_polymer_entity_annotation.annotation_id:PF08798%20AND%20rcsb_polymer_entity_annotation.type:Pfam);[PDBe](https://www.ebi.ac.uk/pdbe/entry/search/index?pfam_accession:PF08798);[PDBj](https://pdbj.org/search/pdb?other_db_select=PFam&other_db_field=PF08798)[PDBsum](/wiki/PDBsum)

[structure summary](https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/GetPfamStr.pl?pfam_id=PF08798)The cas genes in the adaptor and effector modules of the CRISPR-Cas system are believed to have evolved from two different ancestral modules. A [transposon](/wiki/Transposon)-like element called [casposon](/w/index.php?title=Casposon&action=edit&redlink=1) encoding the Cas1-like integrase and potentially other components of the adaptation module was inserted next to the ancestral effector module, which likely functioned as an independent innate immune system. [130] The highly conserved cas1 and cas2 genes of the adaptor module evolved from the ancestral module while a variety of class 1 effector cas genes evolved from the ancestral effector module.

The evolution of these various class 1 effector module cas genes was guided by various mechanisms, such as duplication events.

[[131]](#cite_note-133)On the other hand, each type of class 2 effector module arose from subsequent independent insertions of mobile genetic elements.

[[132]](#cite_note-134)These mobile genetic elements took the place of the multiple gene effector modules to create single gene effector modules that produce large proteins which perform all the necessary tasks of the effector module.

[[133]](#cite_note-Shmakov-2017-135)The spacer regions of CRISPR-Cas systems are taken directly from foreign mobile genetic elements and thus their long-term evolution is hard to trace.

[[133]](#cite_note-Shmakov-2017-135)The non-random evolution of these spacer regions has been found to be highly dependent on the environment and the particular foreign mobile genetic elements it contains.

[[134]](#cite_note-136)

[[135]](#cite_note-137)CRISPR-Cas can immunize bacteria against certain phages and thus halt transmission. For this reason, [Koonin](/wiki/Eugene_Koonin) described CRISPR-Cas as a [Lamarckian](/wiki/Lamarckism) inheritance mechanism. [136] However, this was disputed by a critic who noted, "We should remember [Lamarck] for the good he contributed to science, not for things that resemble his theory only superficially. Indeed, thinking of CRISPR and other phenomena as Lamarckian only obscures the simple and elegant way evolution really works".

But as more recent studies have been conducted, it has become apparent that the acquired spacer regions of CRISPR-Cas systems are indeed a form of Lamarckian evolution because they are genetic mutations that are acquired and then passed on.

[[137]](#cite_note-139)On the other hand, the evolution of the Cas gene machinery that facilitates the system evolves through classic Darwinian evolution.

[[138]](#cite_note-Koonin-2016-140)

[[138]](#cite_note-Koonin-2016-140)### Coevolution

[[edit](/w/index.php?title=CRISPR&action=edit§ion=18)]

Analysis of CRISPR sequences revealed [coevolution](/wiki/Coevolution) of host and viral genomes.[[139]](#cite_note-141)

The basic model of CRISPR evolution is newly incorporated spacers driving phages to mutate their genomes to avoid the bacterial immune response, creating diversity in both the phage and host populations. To resist a phage infection, the sequence of the CRISPR spacer must correspond perfectly to the sequence of the target phage gene. Phages can continue to infect their hosts' given point mutations in the spacer. [129] Similar stringency is required in PAM or the bacterial strain remains phage sensitive.


[[102]](#cite_note-pmid18065545-104)

[[129]](#cite_note-pmid20072129-131)### Rates

[[edit](/w/index.php?title=CRISPR&action=edit§ion=19)]

A study of 124 *S. thermophilus* strains showed that 26% of all spacers were unique and that different CRISPR loci showed different rates of spacer acquisition. [101] Some CRISPR loci evolve more rapidly than others, which allowed the strains' phylogenetic relationships to be determined. A

[comparative genomic](/wiki/Comparative_genomics)analysis showed that

*E. coli*and

*evolve much more slowly than*

[S. enterica](/wiki/Salmonella_enterica)*S. thermophilus*. The latter's strains that diverged 250,000 years ago still contained the same spacer complement.


[[140]](#cite_note-pmid20559554-142)[Metagenomic](/wiki/Metagenomics) analysis of two acid-mine-drainage [biofilms](/wiki/Biofilm) showed that one of the analyzed CRISPRs contained extensive deletions and spacer additions versus the other biofilm, suggesting a higher phage activity/prevalence in one community than the other. [58] In the oral cavity, a temporal study determined that 7–22% of spacers were shared over 17 months within an individual while less than 2% were shared across individuals.


[[110]](#cite_note-pmid21149389-112)From the same environment, a single strain was tracked using [PCR](/wiki/Polymerase_chain_reaction) primers specific to its CRISPR system. Broad-level results of spacer presence/absence showed significant diversity. However, this CRISPR added three spacers over 17 months, [110] suggesting that even in an environment with significant CRISPR diversity some loci evolve slowly.

CRISPRs were analysed from the metagenomes produced for the [Human Microbiome Project](/wiki/Human_Microbiome_Project). [141] Although most were body-site specific, some within a body site are widely shared among individuals. One of these loci originated from

[streptococcal](/wiki/Streptococcal)species and contained ≈15,000 spacers, 50% of which were unique. Similar to the targeted studies of the oral cavity, some showed little evolution over time.


[[141]](#cite_note-pmid22719260-143)CRISPR evolution was studied in [chemostats](/wiki/Chemostat) using *S. thermophilus* to directly examine spacer acquisition rates. In one week, *S. thermophilus* strains acquired up to three spacers when challenged with a single phage. [142] During the same interval, the phage developed

[single-nucleotide polymorphisms](/wiki/Single-nucleotide_polymorphism)that became fixed in the population, suggesting that targeting had prevented phage replication absent these mutations.


[[142]](#cite_note-pmid23057534-144)Another *S. thermophilus* experiment showed that phages can infect and replicate in hosts that have only one targeting spacer. Yet another showed that sensitive hosts can exist in environments with high-phage titres. [143] The chemostat and observational studies suggest many nuances to CRISPR and phage (co)evolution.

## Identification

[[edit](/w/index.php?title=CRISPR&action=edit§ion=20)]

CRISPRs are widely distributed among bacteria and archaea [68] and show some sequence similarities.

Their most notable characteristic is their repeating spacers and direct repeats. This characteristic makes CRISPRs easily identifiable in long sequences of DNA, since the number of repeats decreases the likelihood of a false positive match.

[[116]](#cite_note-pmid17442114-118)

[[144]](#cite_note-146)Analysis of CRISPRs in metagenomic data is more challenging, as CRISPR loci do not typically assemble, due to their repetitive nature or through strain variation, which confuses assembly algorithms. Where many reference genomes are available, [polymerase chain reaction](/wiki/Polymerase_chain_reaction) (PCR) can be used to amplify CRISPR arrays and analyse spacer content.[[101]](#cite_note-pmid18065539-103)[[110]](#cite_note-pmid21149389-112)[[145]](#cite_note-pmid22583485-147)[[146]](#cite_note-pmid23701169-148)[[147]](#cite_note-pmid20927396-149) [148] However, this approach yields information only for specifically targeted CRISPRs and for organisms with sufficient representation in public databases to design reliable polymerase PCR primers. Degenerate repeat-specific primers can be used to amplify CRISPR spacers directly from environmental samples; amplicons containing two or three spacers can be then computationally assembled to reconstruct long CRISPR arrays.


[[148]](#cite_note-pmid31729390-150)The alternative is to extract and reconstruct CRISPR arrays from shotgun metagenomic data. This is computationally more difficult, particularly with second generation sequencing technologies (e.g. 454, Illumina), as the short read lengths prevent more than two or three repeat units appearing in a single read. CRISPR identification in raw reads has been achieved using purely *de novo* identification [149] or by using direct repeat sequences in partially assembled CRISPR arrays from

[contigs](/wiki/Contig)(overlapping DNA segments that together represent a consensus region of DNA)

and direct repeat sequences from published genomes

[[141]](#cite_note-pmid22719260-143)as a hook for identifying direct repeats in individual reads.

[[150]](#cite_note-pmid22732228-152)## Use by phages

[[edit](/w/index.php?title=CRISPR&action=edit§ion=21)]

Another way for bacteria to defend against phage infection is by having [chromosomal islands](/wiki/Genomic_island). A subtype of chromosomal islands called phage-inducible chromosomal island (PICI) is excised from a bacterial chromosome upon phage infection and can inhibit phage replication. [151] PICIs are induced, excised, replicated, and finally packaged into small capsids by certain staphylococcal temperate phages. PICIs use several mechanisms to block phage reproduction. In the first mechanism, PICI-encoded Ppi differentially blocks phage maturation by binding or interacting specifically with phage TerS, hence blocking phage TerS/TerL complex formation responsible for phage DNA packaging. In the second mechanism PICI CpmAB redirects the phage capsid morphogenetic protein to make 95% of SaPI-sized capsid and phage DNA can package only 1/3rd of their genome in these small capsids and hence become nonviable phage.

The third mechanism involves two proteins, PtiA and PtiB, that target the LtrC, which is responsible for the production of virion and lysis proteins. This interference mechanism is modulated by a modulatory protein, PtiM, binds to one of the interference-mediating proteins, PtiA, and hence achieves the required level of interference.

[[152]](#cite_note-154)

[[153]](#cite_note-155)One study showed that lytic ICP1 phage, which specifically targets [Vibrio cholerae](/wiki/Vibrio_cholerae)[serogroup](/wiki/Serogroup) O1, has acquired a CRISPR-Cas system that targets a *V. cholera* PICI-like element. The system has 2 CRISPR loci and 9 Cas genes. It seems to be [homologous](/wiki/Homology_(biology)) to the I-F system found in * Yersinia pestis*. Moreover, like the bacterial CRISPR-Cas system, ICP1 CRISPR-Cas can acquire new sequences, which allows phage and host to co-evolve.


[[154]](#cite_note-pmid23446421-156)

[[155]](#cite_note-Boyd-157)Certain archaeal viruses were shown to carry mini-CRISPR arrays containing one or two spacers. It has been shown that spacers within the virus-borne CRISPR arrays target other viruses and plasmids, suggesting that mini-CRISPR arrays represent a mechanism of heterotypic superinfection exclusion and participate in interviral conflicts.[[148]](#cite_note-pmid31729390-150)

## Applications

[[edit](/w/index.php?title=CRISPR&action=edit§ion=22)]

CRISPR gene editing is a revolutionary technology that allows for precise, targeted modifications to the DNA of living organisms. Developed from a natural defense mechanism found in bacteria, CRISPR-Cas9 is the most commonly used system. Gene editing with CRISPR-Cas9 involves a [Cas9 nuclease](/wiki/Cas9) and an engineered [guide RNA](/wiki/Guide_RNA), which come together to allow for the precise "cutting" of one or both strands of DNA at specific locations within the genome. [156] It makes use of the cell's natural DNA repair systems, including

[non-homologous end joining](/wiki/Non-homologous_end_joining),

[homology-directed repair](/wiki/Homology_directed_repair), or

[mismatch repair](/wiki/DNA_mismatch_repair), to modify, insert, or delete genetic material at these specific cut sites.


[[156]](#cite_note-Koblan_2020-158)This technology has transformed fields such as genetics, medicine,

[[157]](#cite_note-159)

[[158]](#cite_note-Feng_2024-160)

[[159]](#cite_note-Li_2024-161)and agriculture,

[[160]](#cite_note-Kaur_2024-162)offering potential treatments for genetic disorders, advancements in crop engineering, and research into the fundamental workings of life. However, its ethical implications and potential unintended consequences have sparked significant debate.

[[161]](#cite_note-Mishra_2024-163)

[[162]](#cite_note-Shinwari_2018-164)

[[163]](#cite_note-Kozan_2024-165)## See also

[[edit](/w/index.php?title=CRISPR&action=edit§ion=23)]

[CRISPR activation](/wiki/CRISPR_activation)[Anti-CRISPR](/wiki/Anti-CRISPR)[CRISPR/Cas Tools](/wiki/CRISPR/Cas_Tools)[CRISPR gene editing](/wiki/CRISPR_gene_editing)[The CRISPR Journal](/wiki/The_CRISPR_Journal)- "
[Designer baby](/wiki/Designer_baby)" [DRACO](/wiki/DRACO)[Gene knockout](/wiki/Gene_knockout)[Genome-wide CRISPR-Cas9 knockout screens](/wiki/Genome-wide_CRISPR-Cas9_knockout_screens)[Glossary of genetics](/wiki/Glossary_of_genetics)[Human germline engineering](/wiki/Human_germline_engineering)*Human Nature*(2019 documentary film)[MAGESTIC](/wiki/MAGESTIC)*New*eugenics[Prime editing](/wiki/Prime_editing)[RNAi](/wiki/RNAi)[SiRNA](/wiki/SiRNA)[Surveyor nuclease assay](/wiki/Surveyor_nuclease_assay)[Synthetic biology](/wiki/Synthetic_biology)[Zinc finger](/wiki/Zinc_finger)

## Notes

[[edit](/w/index.php?title=CRISPR&action=edit§ion=24)]

## References

[[edit](/w/index.php?title=CRISPR&action=edit§ion=25)]

[^](#cite_ref-Mulepati_2014_1-0)[PDB](/wiki/Protein_Data_Bank):[4QYZ](https://www.rcsb.org/structure/4QYZ): Mulepati S, Héroux A, Bailey S (2014).["Crystal structure of a CRISPR RNA–guided surveillance complex bound to a ssDNA target"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4427192).*Science*.**345**(6203): 1479–1484.[Bibcode](/wiki/Bibcode_(identifier)):[2014Sci...345.1479M](https://ui.adsabs.harvard.edu/abs/2014Sci...345.1479M).[doi](/wiki/Doi_(identifier)):[10.1126/science.1256996](https://doi.org/10.1126%2Fscience.1256996).[PMC](/wiki/PMC_(identifier))[4427192](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4427192).[PMID](/wiki/PMID_(identifier))[25123481](https://pubmed.ncbi.nlm.nih.gov/25123481).- ^
**a**Horvath P,**b**[Barrangou R](/wiki/Rodolphe_Barrangou)(January 2010). "CRISPR/Cas, the immune system of bacteria and archaea".*Science*.**327**(5962): 167–170.[Bibcode](/wiki/Bibcode_(identifier)):[2010Sci...327..167H](https://ui.adsabs.harvard.edu/abs/2010Sci...327..167H).[doi](/wiki/Doi_(identifier)):[10.1126/science.1179555](https://doi.org/10.1126%2Fscience.1179555).[PMID](/wiki/PMID_(identifier))[20056882](https://pubmed.ncbi.nlm.nih.gov/20056882). - ^
**a****b**[Barrangou R](/wiki/Rodolphe_Barrangou)(2015). "The roles of CRISPR-Cas systems in adaptive immunity and beyond".*Current Opinion in Immunology*.**32**: 36–41.[doi](/wiki/Doi_(identifier)):[10.1016/j.coi.2014.12.008](https://doi.org/10.1016%2Fj.coi.2014.12.008).[PMID](/wiki/PMID_(identifier))[25574773](https://pubmed.ncbi.nlm.nih.gov/25574773). - ^
**a****b**Hille F, Richter H, Wong SP, Bratovič M, Ressel S, Charpentier E (March 2018). "The Biology of CRISPR-Cas: Backward and Forward".**c***Cell*.**172**(6): 1239–1259.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2017.11.032](https://doi.org/10.1016%2Fj.cell.2017.11.032).[hdl](/wiki/Hdl_(identifier)):[21.11116/0000-0003-FC0D-4](https://hdl.handle.net/21.11116%2F0000-0003-FC0D-4).[PMID](/wiki/PMID_(identifier))[29522745](https://pubmed.ncbi.nlm.nih.gov/29522745). - ^
**a****b**Rath D, Amlinger L, Rath A, Lundgren M (October 2015).**c**["The CRISPR-Cas immune system: biology, mechanisms and applications"](https://doi.org/10.1016%2Fj.biochi.2015.03.025).*Biochimie*.**117**: 119–128.[doi](/wiki/Doi_(identifier)):[10.1016/j.biochi.2015.03.025](https://doi.org/10.1016%2Fj.biochi.2015.03.025).[PMID](/wiki/PMID_(identifier))[25868999](https://pubmed.ncbi.nlm.nih.gov/25868999). - ^
**a**Redman M, King A, Watson C, King D (August 2016).**b**["What is CRISPR/Cas9?"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4975809).*Archives of Disease in Childhood: Education and Practice Edition*.**101**(4): 213–215.[doi](/wiki/Doi_(identifier)):[10.1136/archdischild-2016-310459](https://doi.org/10.1136%2Farchdischild-2016-310459).[PMC](/wiki/PMC_(identifier))[4975809](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4975809).[PMID](/wiki/PMID_(identifier))[27059283](https://pubmed.ncbi.nlm.nih.gov/27059283). - ^
**a****b**[Barrangou R](/wiki/Rodolphe_Barrangou), Fremaux C, Deveau H, Richards M, Boyaval P, Moineau S, et al. (March 2007). "CRISPR provides acquired resistance against viruses in prokaryotes".*Science*.**315**(5819): 1709–1712.[Bibcode](/wiki/Bibcode_(identifier)):[2007Sci...315.1709B](https://ui.adsabs.harvard.edu/abs/2007Sci...315.1709B).[doi](/wiki/Doi_(identifier)):[10.1126/science.1138140](https://doi.org/10.1126%2Fscience.1138140).[hdl](/wiki/Hdl_(identifier)):[20.500.11794/38902](https://hdl.handle.net/20.500.11794%2F38902).[PMID](/wiki/PMID_(identifier))[17379808](https://pubmed.ncbi.nlm.nih.gov/17379808). ([registration required](/wiki/Wikipedia:Verifiability#Access_to_sources)) - ^
**a**Marraffini LA, Sontheimer EJ (December 2008).**b**["CRISPR interference limits horizontal gene transfer in staphylococci by targeting DNA"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2695655).*Science*.**322**(5909): 1843–1845.[Bibcode](/wiki/Bibcode_(identifier)):[2008Sci...322.1843M](https://ui.adsabs.harvard.edu/abs/2008Sci...322.1843M).[doi](/wiki/Doi_(identifier)):[10.1126/science.1165771](https://doi.org/10.1126%2Fscience.1165771).[PMC](/wiki/PMC_(identifier))[2695655](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2695655).[PMID](/wiki/PMID_(identifier))[19095942](https://pubmed.ncbi.nlm.nih.gov/19095942). Bak RO, Gomez-Ospina N, Porteus MH (August 2018). "Gene Editing on Center Stage".[^](#cite_ref-9)*Trends in Genetics*.**34**(8): 600–611.[doi](/wiki/Doi_(identifier)):[10.1016/j.tig.2018.05.004](https://doi.org/10.1016%2Fj.tig.2018.05.004).[PMID](/wiki/PMID_(identifier))[29908711](https://pubmed.ncbi.nlm.nih.gov/29908711).Zhang F, Wen Y, Guo X (2014).[^](#cite_ref-pmid24651067_10-0)["CRISPR/Cas9 for genome editing: progress, implications and challenges"](https://doi.org/10.1093%2Fhmg%2Fddu125).*Human Molecular Genetics*.**23**(R1): R40–6.[doi](/wiki/Doi_(identifier)):[10.1093/hmg/ddu125](https://doi.org/10.1093%2Fhmg%2Fddu125).[PMID](/wiki/PMID_(identifier))[24651067](https://pubmed.ncbi.nlm.nih.gov/24651067).- ^
**a**CRISPR-CAS9, TALENS and ZFNS – the battle in gene editing**b**[https://www.ptglab.com/news/blog/crispr-cas9-talens-and-zfns-the-battle-in-gene-editing/](https://www.ptglab.com/news/blog/crispr-cas9-talens-and-zfns-the-battle-in-gene-editing/)[Archived](https://web.archive.org/web/20210525031855/https://www.ptglab.com/news/blog/crispr-cas9-talens-and-zfns-the-battle-in-gene-editing/)2021-05-25 at the[Wayback Machine](/wiki/Wayback_Machine) - ^
**a****b****c****d**Hsu PD, Lander ES,**e**[Zhang F](/wiki/Feng_Zhang)(June 2014).["Development and applications of CRISPR-Cas9 for genome engineering"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4343198).*Cell*.**157**(6): 1262–1278.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2014.05.010](https://doi.org/10.1016%2Fj.cell.2014.05.010).[PMC](/wiki/PMC_(identifier))[4343198](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4343198).[PMID](/wiki/PMID_(identifier))[24906146](https://pubmed.ncbi.nlm.nih.gov/24906146). - ^
**a****b**["Press release: The Nobel Prize in Chemistry 2020"](https://www.nobelprize.org/prizes/chemistry/2020/press-release/). Nobel Foundation.[Archived](https://web.archive.org/web/20210115172618/https://www.nobelprize.org/prizes/chemistry/2020/press-release/)from the original on 15 January 2021. Retrieved 7 October 2020. Wu KJ, Peltier E (7 October 2020).[^](#cite_ref-NYT-20201007_14-0)["Nobel Prize in Chemistry Awarded to 2 Scientists for Work on Genome Editing – Emmanuelle Charpentier and Jennifer A. Doudna developed the Crispr tool, which can alter the DNA of animals, plants and microorganisms with high precision"](https://www.nytimes.com/2020/10/07/science/nobel-prize-chemistry-crispr.html)..[The New York Times](/wiki/The_New_York_Times)[Archived](https://web.archive.org/web/20201008094313/https://www.nytimes.com/2020/10/07/science/nobel-prize-chemistry-crispr.html)from the original on 8 October 2020. Retrieved 7 October 2020.Rawat A, Roy M, Jyoti A, Kaushik S, Verma K, Srivastava VK (August 2021).[^](#cite_ref-Rawat_2021_15-0)["Cysteine proteases: Battling pathogenic parasitic protozoans with omnipresent enzymes"](https://doi.org/10.1016%2Fj.micres.2021.126784).*Microbiological Research*.**249**126784.[doi](/wiki/Doi_(identifier)):[10.1016/j.micres.2021.126784](https://doi.org/10.1016%2Fj.micres.2021.126784).[PMID](/wiki/PMID_(identifier))[33989978](https://pubmed.ncbi.nlm.nih.gov/33989978).- ^
**a**Ishino Y, Shinagawa H, Makino K, Amemura M, Nakata A (December 1987).**b**["Nucleotide sequence of the iap gene, responsible for alkaline phosphatase isozyme conversion in Escherichia coli, and identification of the gene product"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC213968).*Journal of Bacteriology*.**169**(12): 5429–5433.[doi](/wiki/Doi_(identifier)):[10.1128/jb.169.12.5429-5433.1987](https://doi.org/10.1128%2Fjb.169.12.5429-5433.1987).[PMC](/wiki/PMC_(identifier))[213968](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC213968).[PMID](/wiki/PMID_(identifier))[3316184](https://pubmed.ncbi.nlm.nih.gov/3316184). van Soolingen D, de Haas PE, Hermans PW, Groenen PM, van Embden JD (August 1993).[^](#cite_ref-Soolingen1993_17-0)["Comparison of various repetitive DNA elements as genetic markers for strain differentiation and epidemiology of Mycobacterium tuberculosis"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC265684).*Journal of Clinical Microbiology*.**31**(8): 1987–1995.[doi](/wiki/Doi_(identifier)):[10.1128/JCM.31.8.1987-1995.1993](https://doi.org/10.1128%2FJCM.31.8.1987-1995.1993).[PMC](/wiki/PMC_(identifier))[265684](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC265684).[PMID](/wiki/PMID_(identifier))[7690367](https://pubmed.ncbi.nlm.nih.gov/7690367).Groenen PM, Bunschoten AE, van Soolingen D, van Embden JD (December 1993). "Nature of DNA polymorphism in the direct repeat cluster of Mycobacterium tuberculosis; application for strain differentiation by a novel typing method".[^](#cite_ref-Groenen1993_18-0)*Molecular Microbiology*.**10**(5): 1057–1065.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.1993.tb00976.x](https://doi.org/10.1111%2Fj.1365-2958.1993.tb00976.x).[PMID](/wiki/PMID_(identifier))[7934856](https://pubmed.ncbi.nlm.nih.gov/7934856).- ^
**a****b**Mojica FJ, Montoliu L (2016). "On the Origin of CRISPR-Cas Technology: From Prokaryotes to Mammals".**c***Trends in Microbiology*.**24**(10): 811–820.[doi](/wiki/Doi_(identifier)):[10.1016/j.tim.2016.06.005](https://doi.org/10.1016%2Fj.tim.2016.06.005).[PMID](/wiki/PMID_(identifier))[27401123](https://pubmed.ncbi.nlm.nih.gov/27401123). - ^
**a**Mojica FJ, Rodriguez-Valera F (2016). "The discovery of CRISPR in archaea and bacteria".**b***The FEBS Journal*.**283**(17): 3162–3169.[doi](/wiki/Doi_(identifier)):[10.1111/febs.13766](https://doi.org/10.1111%2Ffebs.13766).[hdl](/wiki/Hdl_(identifier)):[10045/57676](https://hdl.handle.net/10045%2F57676).[PMID](/wiki/PMID_(identifier))[27234458](https://pubmed.ncbi.nlm.nih.gov/27234458). Mojica FJ, Díez-Villaseñor C, Soria E, Juez G (April 2000).[^](#cite_ref-Mojica_2000_21-0)["Biological significance of a family of regularly spaced repeats in the genomes of Archaea, Bacteria and mitochondria"](https://doi.org/10.1046%2Fj.1365-2958.2000.01838.x).*Molecular Microbiology*.**36**(1): 244–246.[doi](/wiki/Doi_(identifier)):[10.1046/j.1365-2958.2000.01838.x](https://doi.org/10.1046%2Fj.1365-2958.2000.01838.x).[PMID](/wiki/PMID_(identifier))[10760181](https://pubmed.ncbi.nlm.nih.gov/10760181).[^](#cite_ref-Isaacson_2021_22-0)[Isaacson W](/wiki/Walter_Isaacson)(2021).. New York: Simon & Schuster. p. 73.*The Code Breaker: Jennifer Doudna, Gene Editing, and the Future of the Human Race*[ISBN](/wiki/ISBN_(identifier))[978-1-9821-1585-2](/wiki/Special:BookSources/978-1-9821-1585-2).[OCLC](/wiki/OCLC_(identifier))[1239982737](https://search.worldcat.org/oclc/1239982737).[Archived](https://web.archive.org/web/20230114155931/https://books.google.com/books?id=f_D3DwAAQBAJ)from the original on 2023-01-14. Retrieved 2021-10-20.[^](#cite_ref-Barrangou_2013_23-0)[Barrangou R](/wiki/Rodolphe_Barrangou), van der Oost J (2013).*CRISPR-Cas Systems: RNA-mediated Adaptive Immunity in Bacteria and Archaea*. Heidelberg: Springer. p. 6.[ISBN](/wiki/ISBN_(identifier))[978-3-642-34656-9](/wiki/Special:BookSources/978-3-642-34656-9).Tang TH, Bachellerie JP, Rozhdestvensky T, Bortolin ML, Huber H, Drungowski M, et al. (May 2002).[^](#cite_ref-pmid12032318_24-0)["Identification of 86 candidates for small non-messenger RNAs from the archaeon Archaeoglobus fulgidus"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC124276).*Proceedings of the National Academy of Sciences of the United States of America*.**99**(11): 7536–7541.[Bibcode](/wiki/Bibcode_(identifier)):[2002PNAS...99.7536T](https://ui.adsabs.harvard.edu/abs/2002PNAS...99.7536T).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.112047299](https://doi.org/10.1073%2Fpnas.112047299).[PMC](/wiki/PMC_(identifier))[124276](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC124276).[PMID](/wiki/PMID_(identifier))[12032318](https://pubmed.ncbi.nlm.nih.gov/12032318).Charpentier E, Richter H, van der Oost J, White MF (May 2015).[^](#cite_ref-pmid25994611_25-0)["Biogenesis pathways of RNA guides in archaeal and bacterial CRISPR-Cas adaptive immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5965381).*FEMS Microbiology Reviews*.**39**(3): 428–441.[doi](/wiki/Doi_(identifier)):[10.1093/femsre/fuv023](https://doi.org/10.1093%2Ffemsre%2Ffuv023).[PMC](/wiki/PMC_(identifier))[5965381](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5965381).[PMID](/wiki/PMID_(identifier))[25994611](https://pubmed.ncbi.nlm.nih.gov/25994611).Romero DA, Magill D, Millen A, Horvath P, Fremaux C (November 2020). "Dairy lactococcal and streptococcal phage-host interactions: an industrial perspective in an evolving phage landscape".[^](#cite_ref-Romero_2020_26-0)*FEMS Microbiology Reviews*.**44**(6): 909–932.[doi](/wiki/Doi_(identifier)):[10.1093/femsre/fuaa048](https://doi.org/10.1093%2Ffemsre%2Ffuaa048).[PMID](/wiki/PMID_(identifier))[33016324](https://pubmed.ncbi.nlm.nih.gov/33016324).Molteni M, Huckins G (1 August 2020).[^](#cite_ref-rbpont_27-0)["The WIRED Guide to Crispr"](https://www.wired.com/story/wired-guide-to-crispr/). Condé Nast. Wired Magazine.[Archived](https://web.archive.org/web/20211023062846/https://www.wired.com/story/wired-guide-to-crispr/)from the original on 23 October 2021. Retrieved 23 February 2021.Jansen R, Embden JD, Gaastra W, Schouls LM (March 2002).[^](#cite_ref-pmid11952905_28-0)["Identification of genes that are associated with DNA repeats in prokaryotes"](https://doi.org/10.1046%2Fj.1365-2958.2002.02839.x).*Molecular Microbiology*.**43**(6): 1565–1575.[doi](/wiki/Doi_(identifier)):[10.1046/j.1365-2958.2002.02839.x](https://doi.org/10.1046%2Fj.1365-2958.2002.02839.x).[PMID](/wiki/PMID_(identifier))[11952905](https://pubmed.ncbi.nlm.nih.gov/11952905).- ^
**a**Horvath P,**b**[Barrangou R](/wiki/Rodolphe_Barrangou)(January 2010). "CRISPR/Cas, the immune system of bacteria and archaea".*Science*.**327**(5962): 167–170.[Bibcode](/wiki/Bibcode_(identifier)):[2010Sci...327..167H](https://ui.adsabs.harvard.edu/abs/2010Sci...327..167H).[doi](/wiki/Doi_(identifier)):[10.1126/Science.1179555](https://doi.org/10.1126%2FScience.1179555).[PMID](/wiki/PMID_(identifier))[20056882](https://pubmed.ncbi.nlm.nih.gov/20056882). - ^
**a**Marraffini LA, Sontheimer EJ (March 2010).**b**["CRISPR interference: RNA-directed adaptive immunity in bacteria and archaea"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928866).*Nature Reviews Genetics*.**11**(3): 181–190.[doi](/wiki/Doi_(identifier)):[10.1038/nrg2749](https://doi.org/10.1038%2Fnrg2749).[PMC](/wiki/PMC_(identifier))[2928866](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928866).[PMID](/wiki/PMID_(identifier))[20125085](https://pubmed.ncbi.nlm.nih.gov/20125085). Grissa I, Vergnaud G, Pourcel C (May 2007).[^](#cite_ref-pmid17521438_31-0)["The CRISPRdb database and tools to display CRISPRs and to generate dictionaries of spacers and repeats"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1892036).*BMC Bioinformatics*.**8**172.[doi](/wiki/Doi_(identifier)):[10.1186/1471-2105-8-172](https://doi.org/10.1186%2F1471-2105-8-172).[PMC](/wiki/PMC_(identifier))[1892036](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1892036).[PMID](/wiki/PMID_(identifier))[17521438](https://pubmed.ncbi.nlm.nih.gov/17521438).- ^
**a**Pourcel C, Salvignol G, Vergnaud G (March 2005).**b**["CRISPR elements in Yersinia pestis acquire new repeats by preferential uptake of bacteriophage DNA, and provide additional tools for evolutionary studies"](https://doi.org/10.1099%2Fmic.0.27437-0).*Microbiology*.**151**(Pt 3): 653–663.[doi](/wiki/Doi_(identifier)):[10.1099/mic.0.27437-0](https://doi.org/10.1099%2Fmic.0.27437-0).[PMID](/wiki/PMID_(identifier))[15758212](https://pubmed.ncbi.nlm.nih.gov/15758212). - ^
**a**Mojica FJ, Díez-Villaseñor C, García-Martínez J, Soria E (February 2005). "Intervening sequences of regularly spaced prokaryotic repeats derive from foreign genetic elements".**b***Journal of Molecular Evolution*.**60**(2): 174–182.[Bibcode](/wiki/Bibcode_(identifier)):[2005JMolE..60..174M](https://ui.adsabs.harvard.edu/abs/2005JMolE..60..174M).[doi](/wiki/Doi_(identifier)):[10.1007/s00239-004-0046-3](https://doi.org/10.1007%2Fs00239-004-0046-3).[PMID](/wiki/PMID_(identifier))[15791728](https://pubmed.ncbi.nlm.nih.gov/15791728). - ^
**a**Bolotin A, Quinquis B, Sorokin A, Ehrlich SD (August 2005).**b**["Clustered regularly interspaced short palindrome repeats (CRISPRs) have spacers of extrachromosomal origin"](https://doi.org/10.1099%2Fmic.0.28048-0).*Microbiology*.**151**(Pt 8): 2551–2561.[doi](/wiki/Doi_(identifier)):[10.1099/mic.0.28048-0](https://doi.org/10.1099%2Fmic.0.28048-0).[PMID](/wiki/PMID_(identifier))[16079334](https://pubmed.ncbi.nlm.nih.gov/16079334). Morange M (June 2015).[^](#cite_ref-pmid25963251_35-0)["What history tells us XXXVII. CRISPR-Cas: The discovery of an immune system in prokaryotes"](https://doi.org/10.1007%2Fs12038-015-9532-6).*Journal of Biosciences*.**40**(2): 221–223.[doi](/wiki/Doi_(identifier)):[10.1007/s12038-015-9532-6](https://doi.org/10.1007%2Fs12038-015-9532-6).[PMID](/wiki/PMID_(identifier))[25963251](https://pubmed.ncbi.nlm.nih.gov/25963251).Lander ES (January 2016).[^](#cite_ref-Lander_2016_36-0)["The Heroes of CRISPR"](https://doi.org/10.1016%2Fj.cell.2015.12.041).*Cell*.**164**(1–2): 18–28.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2015.12.041](https://doi.org/10.1016%2Fj.cell.2015.12.041).[PMID](/wiki/PMID_(identifier))[26771483](https://pubmed.ncbi.nlm.nih.gov/26771483).Makarova KS, Grishin NV, Shabalina SA, Wolf YI,[^](#cite_ref-Makarova2006_37-0)[Koonin EV](/wiki/Eugene_Koonin)(March 2006).["A putative RNA-interference-based immune system in prokaryotes: computational analysis of the predicted enzymatic machinery, functional analogies with eukaryotic RNAi, and hypothetical mechanisms of action"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1462988).*Biology Direct*.**1**7.[doi](/wiki/Doi_(identifier)):[10.1186/1745-6150-1-7](https://doi.org/10.1186%2F1745-6150-1-7).[PMC](/wiki/PMC_(identifier))[1462988](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1462988).[PMID](/wiki/PMID_(identifier))[16545108](https://pubmed.ncbi.nlm.nih.gov/16545108).[^](#cite_ref-craze_38-0)[Pennisi E](/wiki/Elizabeth_Pennisi)(August 2013). "The CRISPR craze". News Focus.*Science*.**341**(6148): 833–836.[Bibcode](/wiki/Bibcode_(identifier)):[2013Sci...341..833P](https://ui.adsabs.harvard.edu/abs/2013Sci...341..833P).[doi](/wiki/Doi_(identifier)):[10.1126/science.341.6148.833](https://doi.org/10.1126%2Fscience.341.6148.833).[PMID](/wiki/PMID_(identifier))[23970676](https://pubmed.ncbi.nlm.nih.gov/23970676).- ^
**a****b**Marraffini LA (October 2015). "CRISPR-Cas immunity in prokaryotes".**c***Nature*.**526**(7571): 55–61.[Bibcode](/wiki/Bibcode_(identifier)):[2015Natur.526...55M](https://ui.adsabs.harvard.edu/abs/2015Natur.526...55M).[doi](/wiki/Doi_(identifier)):[10.1038/nature15386](https://doi.org/10.1038%2Fnature15386).[PMID](/wiki/PMID_(identifier))[26432244](https://pubmed.ncbi.nlm.nih.gov/26432244). Brouns SJ, Jore MM, Lundgren M, Westra ER, Slijkhuis RJ, Snijders AP, et al. (August 2008).[^](#cite_ref-Brouns_2008_40-0)["Small CRISPR RNAs guide antiviral defense in prokaryotes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5898235).*Science*.**321**(5891): 960–964.[Bibcode](/wiki/Bibcode_(identifier)):[2008Sci...321..960B](https://ui.adsabs.harvard.edu/abs/2008Sci...321..960B).[doi](/wiki/Doi_(identifier)):[10.1126/science.1159689](https://doi.org/10.1126%2Fscience.1159689).[PMC](/wiki/PMC_(identifier))[5898235](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5898235).[PMID](/wiki/PMID_(identifier))[18703739](https://pubmed.ncbi.nlm.nih.gov/18703739).- ^
**a**Garneau JE, Dupuis MÈ, Villion M, Romero DA,**b**[Barrangou R](/wiki/Rodolphe_Barrangou), Boyaval P, et al. (November 2010). "The CRISPR/Cas bacterial immune system cleaves bacteriophage and plasmid DNA".*Nature*.**468**(7320): 67–71.[Bibcode](/wiki/Bibcode_(identifier)):[2010Natur.468...67G](https://ui.adsabs.harvard.edu/abs/2010Natur.468...67G).[doi](/wiki/Doi_(identifier)):[10.1038/nature09523](https://doi.org/10.1038%2Fnature09523).[PMID](/wiki/PMID_(identifier))[21048762](https://pubmed.ncbi.nlm.nih.gov/21048762). Jinek M, Chylinski K, Fonfara I, Hauer M, Doudna JA, Charpentier E (August 2012).[^](#cite_ref-P22745249_42-0)["A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286148).*Science*.**337**(6096): 816–821.[Bibcode](/wiki/Bibcode_(identifier)):[2012Sci...337..816J](https://ui.adsabs.harvard.edu/abs/2012Sci...337..816J).[doi](/wiki/Doi_(identifier)):[10.1126/science.1225829](https://doi.org/10.1126%2Fscience.1225829).[PMC](/wiki/PMC_(identifier))[6286148](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286148).[PMID](/wiki/PMID_(identifier))[22745249](https://pubmed.ncbi.nlm.nih.gov/22745249).Mojica FJ, Montoliu L (2016). "On the Origin of CRISPR-Cas Technology: From Prokaryotes to Mammals".[^](#cite_ref-43)*Trends in Microbiology*.**24**(10): 811–820.[doi](/wiki/Doi_(identifier)):[10.1016/j.tim.2016.06.005](https://doi.org/10.1016%2Fj.tim.2016.06.005).[PMID](/wiki/PMID_(identifier))[27401123](https://pubmed.ncbi.nlm.nih.gov/27401123).Cong L, Ran FA, Cox D, Lin S, Barretto R, Habib N, et al. (February 2013).[^](#cite_ref-44)["Multiplex genome engineering using CRISPR/Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795411).*Science*.**339**(6121): 819–823.[Bibcode](/wiki/Bibcode_(identifier)):[2013Sci...339..819C](https://ui.adsabs.harvard.edu/abs/2013Sci...339..819C).[doi](/wiki/Doi_(identifier)):[10.1126/science.1231143](https://doi.org/10.1126%2Fscience.1231143).[PMC](/wiki/PMC_(identifier))[3795411](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795411).[PMID](/wiki/PMID_(identifier))[23287718](https://pubmed.ncbi.nlm.nih.gov/23287718).Mali P, Yang L, Esvelt KM, Aach J, Guell M, DiCarlo JE, et al. (February 2013).[^](#cite_ref-45)["RNA-guided human genome engineering via Cas9"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3712628).*Science*.**339**(6121): 823–826.[Bibcode](/wiki/Bibcode_(identifier)):[2013Sci...339..823M](https://ui.adsabs.harvard.edu/abs/2013Sci...339..823M).[doi](/wiki/Doi_(identifier)):[10.1126/science.1232033](https://doi.org/10.1126%2Fscience.1232033).[PMC](/wiki/PMC_(identifier))[3712628](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3712628).[PMID](/wiki/PMID_(identifier))[23287722](https://pubmed.ncbi.nlm.nih.gov/23287722).Liang P, Xu Y, Zhang X, Ding C, Huang R, Zhang Z, et al. (May 2015).[^](#cite_ref-46)["CRISPR/Cas9-mediated gene editing in human tripronuclear zygotes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4417674).*Protein & Cell*.**6**(5): 363–372.[doi](/wiki/Doi_(identifier)):[10.1007/s13238-015-0153-5](https://doi.org/10.1007%2Fs13238-015-0153-5).[PMC](/wiki/PMC_(identifier))[4417674](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4417674).[PMID](/wiki/PMID_(identifier))[25894090](https://pubmed.ncbi.nlm.nih.gov/25894090).- ^
**a**Deltcheva E, Chylinski K, Sharma CM, Gonzales K, Chao Y, Pirzada ZA, et al. (March 2011).**b**["CRISPR RNA maturation by trans-encoded small RNA and host factor RNase III"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3070239).*Nature*.**471**(7340): 602–607.[Bibcode](/wiki/Bibcode_(identifier)):[2011Natur.471..602D](https://ui.adsabs.harvard.edu/abs/2011Natur.471..602D).[doi](/wiki/Doi_(identifier)):[10.1038/nature09886](https://doi.org/10.1038%2Fnature09886).[PMC](/wiki/PMC_(identifier))[3070239](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3070239).[PMID](/wiki/PMID_(identifier))[21455174](https://pubmed.ncbi.nlm.nih.gov/21455174). [^](#cite_ref-Barrangou2015_48-0)[Barrangou R](/wiki/Rodolphe_Barrangou)(November 2015).["Diversity of CRISPR-Cas immune systems and molecular machines"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4638107).*Genome Biology*.**16**247.[doi](/wiki/Doi_(identifier)):[10.1186/s13059-015-0816-9](https://doi.org/10.1186%2Fs13059-015-0816-9).[PMC](/wiki/PMC_(identifier))[4638107](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4638107).[PMID](/wiki/PMID_(identifier))[26549499](https://pubmed.ncbi.nlm.nih.gov/26549499).Jinek M, Chylinski K, Fonfara I, Hauer M,[^](#cite_ref-pmid22745249_49-0)[Doudna JA](/wiki/Jennifer_Doudna), Charpentier E (August 2012).["A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286148).*Science*.**337**(6096): 816–821.[Bibcode](/wiki/Bibcode_(identifier)):[2012Sci...337..816J](https://ui.adsabs.harvard.edu/abs/2012Sci...337..816J).[doi](/wiki/Doi_(identifier)):[10.1126/science.1225829](https://doi.org/10.1126%2Fscience.1225829).[PMC](/wiki/PMC_(identifier))[6286148](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286148).[PMID](/wiki/PMID_(identifier))[22745249](https://pubmed.ncbi.nlm.nih.gov/22745249).Cong L, Ran FA, Cox D, Lin S, Barretto R, Habib N, et al. (February 2013).[^](#cite_ref-Cong2013_50-0)["Multiplex genome engineering using CRISPR/Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795411).*Science*.**339**(6121): 819–823.[Bibcode](/wiki/Bibcode_(identifier)):[2013Sci...339..819C](https://ui.adsabs.harvard.edu/abs/2013Sci...339..819C).[doi](/wiki/Doi_(identifier)):[10.1126/science.1231143](https://doi.org/10.1126%2Fscience.1231143).[PMC](/wiki/PMC_(identifier))[3795411](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795411).[PMID](/wiki/PMID_(identifier))[23287718](https://pubmed.ncbi.nlm.nih.gov/23287718).Mali P, Yang L, Esvelt KM, Aach J, Guell M, DiCarlo JE, et al. (February 2013).[^](#cite_ref-Mali2013_51-0)["RNA-guided human genome engineering via Cas9"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3712628).*Science*.**339**(6121): 823–826.[Bibcode](/wiki/Bibcode_(identifier)):[2013Sci...339..823M](https://ui.adsabs.harvard.edu/abs/2013Sci...339..823M).[doi](/wiki/Doi_(identifier)):[10.1126/science.1232033](https://doi.org/10.1126%2Fscience.1232033).[PMC](/wiki/PMC_(identifier))[3712628](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3712628).[PMID](/wiki/PMID_(identifier))[23287722](https://pubmed.ncbi.nlm.nih.gov/23287722).Abudayyeh OO, Gootenberg JS, Konermann S, Joung J, Slaymaker IM, Cox DB, et al. (August 2016).[^](#cite_ref-52)["C2c2 is a single-component programmable RNA-guided RNA-targeting CRISPR effector"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5127784).*Science*.**353**(6299) aaf5573.[doi](/wiki/Doi_(identifier)):[10.1126/science.aaf5573](https://doi.org/10.1126%2Fscience.aaf5573).[PMC](/wiki/PMC_(identifier))[5127784](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5127784).[PMID](/wiki/PMID_(identifier))[27256883](https://pubmed.ncbi.nlm.nih.gov/27256883).Gootenberg JS, Abudayyeh OO, Lee JW, Essletzbichler P, Dy AJ, Joung J, et al. (April 2017).[^](#cite_ref-Nucleic_acid_detection_with_CRISPR_53-0)["Nucleic acid detection with CRISPR-Cas13a/C2c2"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5526198).*Science*.**356**(6336): 438–442.[Bibcode](/wiki/Bibcode_(identifier)):[2017Sci...356..438G](https://ui.adsabs.harvard.edu/abs/2017Sci...356..438G).[doi](/wiki/Doi_(identifier)):[10.1126/science.aam9321](https://doi.org/10.1126%2Fscience.aam9321).[PMC](/wiki/PMC_(identifier))[5526198](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5526198).[PMID](/wiki/PMID_(identifier))[28408723](https://pubmed.ncbi.nlm.nih.gov/28408723).Gootenberg JS, Abudayyeh OO, Kellner MJ, Joung J, Collins JJ, Zhang F (April 2018).[^](#cite_ref-54)["Multiplexed and portable nucleic acid detection platform with Cas13, Cas12a, and Csm6"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5961727).*Science*.**360**(6387): 439–444.[Bibcode](/wiki/Bibcode_(identifier)):[2018Sci...360..439G](https://ui.adsabs.harvard.edu/abs/2018Sci...360..439G).[doi](/wiki/Doi_(identifier)):[10.1126/science.aaq0179](https://doi.org/10.1126%2Fscience.aaq0179).[PMC](/wiki/PMC_(identifier))[5961727](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5961727).[PMID](/wiki/PMID_(identifier))[29449508](https://pubmed.ncbi.nlm.nih.gov/29449508).Iwasaki RS, Batey RT (September 2020).[^](#cite_ref-SPRINT:_a_Cas13a-based_platform_for_55-0)["SPRINT: a Cas13a-based platform for detection of small molecules"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7515716).*Nucleic Acids Research*.**48**(17): e101.[doi](/wiki/Doi_(identifier)):[10.1093/nar/gkaa673](https://doi.org/10.1093%2Fnar%2Fgkaa673).[PMC](/wiki/PMC_(identifier))[7515716](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7515716).[PMID](/wiki/PMID_(identifier))[32797156](https://pubmed.ncbi.nlm.nih.gov/32797156).Hille F, Charpentier E (November 2016).[^](#cite_ref-Hille2016_56-0)["CRISPR-Cas: biology, mechanisms and relevance"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5052741).*Philosophical Transactions of the Royal Society of London. Series B, Biological Sciences*.**371**(1707) 20150496.[doi](/wiki/Doi_(identifier)):[10.1098/rstb.2015.0496](https://doi.org/10.1098%2Frstb.2015.0496).[PMC](/wiki/PMC_(identifier))[5052741](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5052741).[PMID](/wiki/PMID_(identifier))[27672148](https://pubmed.ncbi.nlm.nih.gov/27672148).- ^
**a****b****c**[Barrangou R](/wiki/Rodolphe_Barrangou), Marraffini LA (April 2014).["CRISPR-Cas systems: Prokaryotes upgrade to adaptive immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4025954).*Molecular Cell*.**54**(2): 234–244.[doi](/wiki/Doi_(identifier)):[10.1016/j.molcel.2014.03.011](https://doi.org/10.1016%2Fj.molcel.2014.03.011).[PMC](/wiki/PMC_(identifier))[4025954](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4025954).[PMID](/wiki/PMID_(identifier))[24766887](https://pubmed.ncbi.nlm.nih.gov/24766887). - ^
**a****b**Tyson GW, Banfield JF (January 2008). "Rapidly evolving CRISPRs implicated in acquired resistance of microorganisms to viruses".**c***Environmental Microbiology*.**10**(1): 200–207.[Bibcode](/wiki/Bibcode_(identifier)):[2008EnvMi..10..200T](https://ui.adsabs.harvard.edu/abs/2008EnvMi..10..200T).[doi](/wiki/Doi_(identifier)):[10.1111/j.1462-2920.2007.01444.x](https://doi.org/10.1111%2Fj.1462-2920.2007.01444.x).[PMID](/wiki/PMID_(identifier))[17894817](https://pubmed.ncbi.nlm.nih.gov/17894817). - ^
**a**Koonin EV, Makarova KS (May 2019).**b**["Origins and evolution of CRISPR-Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6452270).*Philosophical Transactions of the Royal Society of London. Series B, Biological Sciences*.**374**(1772) 20180087.[doi](/wiki/Doi_(identifier)):[10.1098/rstb.2018.0087](https://doi.org/10.1098%2Frstb.2018.0087).[PMC](/wiki/PMC_(identifier))[6452270](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6452270).[PMID](/wiki/PMID_(identifier))[30905284](https://pubmed.ncbi.nlm.nih.gov/30905284). - ^
**a****b**Wright AV, Nuñez JK,**c**[Doudna JA](/wiki/Jennifer_Doudna)(January 2016).["Biology and Applications of CRISPR Systems: Harnessing Nature's Toolbox for Genome Engineering"](https://doi.org/10.1016%2Fj.cell.2015.12.035).*Cell*.**164**(1–2): 29–44.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2015.12.035](https://doi.org/10.1016%2Fj.cell.2015.12.035).[PMID](/wiki/PMID_(identifier))[26771484](https://pubmed.ncbi.nlm.nih.gov/26771484). Makarova KS, Wolf YI, Iranzo J, Shmakov SA, Alkhnbashi OS, Brouns SJ, et al. (December 2019).[^](#cite_ref-Makarova2020_61-0)["Evolutionary classification of CRISPR–Cas systems: a burst of class 2 and derived variants"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8905525).*Nature Reviews Microbiology*.**18**(1): 67–83.[doi](/wiki/Doi_(identifier)):[10.1038/s41579-019-0299-x](https://doi.org/10.1038%2Fs41579-019-0299-x).[hdl](/wiki/Hdl_(identifier)):[10045/102627](https://hdl.handle.net/10045%2F102627).[PMC](/wiki/PMC_(identifier))[8905525](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8905525).[PMID](/wiki/PMID_(identifier))[31857715](https://pubmed.ncbi.nlm.nih.gov/31857715).- ^
**a****b****c****d****e****f****g**Makarova KS, Wolf YI, Alkhnbashi OS, Costa F, Shah SA, Saunders SJ, et al. (November 2015).**h**["An updated evolutionary classification of CRISPR-Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5426118).*Nature Reviews. Microbiology*.**13**(11): 722–736.[doi](/wiki/Doi_(identifier)):[10.1038/nrmicro3569](https://doi.org/10.1038%2Fnrmicro3569).[PMC](/wiki/PMC_(identifier))[5426118](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5426118).[PMID](/wiki/PMID_(identifier))[26411297](https://pubmed.ncbi.nlm.nih.gov/26411297). - ^
**a****b**Wiedenheft B, Sternberg SH,**c**[Doudna JA](/wiki/Jennifer_Doudna)(February 2012). "RNA-guided genetic silencing systems in bacteria and archaea".*Nature*.**482**(7385): 331–338.[Bibcode](/wiki/Bibcode_(identifier)):[2012Natur.482..331W](https://ui.adsabs.harvard.edu/abs/2012Natur.482..331W).[doi](/wiki/Doi_(identifier)):[10.1038/nature10886](https://doi.org/10.1038%2Fnature10886).[PMID](/wiki/PMID_(identifier))[22337052](https://pubmed.ncbi.nlm.nih.gov/22337052). - ^
**a**Deng L, Garrett RA, Shah SA, Peng X, She Q (March 2013). "A novel interference mechanism by a type IIIB CRISPR-Cmr module in Sulfolobus".**b***Molecular Microbiology*.**87**(5): 1088–1099.[doi](/wiki/Doi_(identifier)):[10.1111/mmi.12152](https://doi.org/10.1111%2Fmmi.12152).[PMID](/wiki/PMID_(identifier))[23320564](https://pubmed.ncbi.nlm.nih.gov/23320564). Sinkunas T, Gasiunas G, Fremaux C,[^](#cite_ref-pmid21343909_65-0)[Barrangou R](/wiki/Rodolphe_Barrangou), Horvath P, Siksnys V (April 2011).["Cas3 is a single-stranded DNA nuclease and ATP-dependent helicase in the CRISPR/Cas immune system"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3094125).*The EMBO Journal*.**30**(7): 1335–1342.[doi](/wiki/Doi_(identifier)):[10.1038/emboj.2011.41](https://doi.org/10.1038%2Femboj.2011.41).[PMC](/wiki/PMC_(identifier))[3094125](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3094125).[PMID](/wiki/PMID_(identifier))[21343909](https://pubmed.ncbi.nlm.nih.gov/21343909).Huo Y, Nam KH, Ding F, Lee H, Wu L, Xiao Y, et al. (September 2014).[^](#cite_ref-ReferenceB_66-0)["Structures of CRISPR Cas3 offer mechanistic insights into Cascade-activated DNA unwinding and degradation"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4156918).*Nature Structural & Molecular Biology*.**21**(9): 771–777.[doi](/wiki/Doi_(identifier)):[10.1038/nsmb.2875](https://doi.org/10.1038%2Fnsmb.2875).[PMC](/wiki/PMC_(identifier))[4156918](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4156918).[PMID](/wiki/PMID_(identifier))[25132177](https://pubmed.ncbi.nlm.nih.gov/25132177).Brendel J, Stoll B, Lange SJ, Sharma K, Lenz C, Stachler AE, et al. (March 2014).[^](#cite_ref-67)["A complex of Cas proteins 5, 6, and 7 is required for the biogenesis and stability of clustered regularly interspaced short palindromic repeats (crispr)-derived rnas (crrnas) in Haloferax volcanii"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3945376).*The Journal of Biological Chemistry*.**289**(10): 7164–77.[doi](/wiki/Doi_(identifier)):[10.1074/jbc.M113.508184](https://doi.org/10.1074%2Fjbc.M113.508184).[PMC](/wiki/PMC_(identifier))[3945376](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3945376).[PMID](/wiki/PMID_(identifier))[24459147](https://pubmed.ncbi.nlm.nih.gov/24459147).- ^
**a**Chylinski K, Makarova KS, Charpentier E,**b**[Koonin EV](/wiki/Eugene_Koonin)(June 2014).["Classification and evolution of type II CRISPR-Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4041416).*Nucleic Acids Research*.**42**(10): 6091–6105.[doi](/wiki/Doi_(identifier)):[10.1093/nar/gku241](https://doi.org/10.1093%2Fnar%2Fgku241).[PMC](/wiki/PMC_(identifier))[4041416](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4041416).[PMID](/wiki/PMID_(identifier))[24728998](https://pubmed.ncbi.nlm.nih.gov/24728998). - ^
**a**Makarova KS, Aravind L, Wolf YI,**b**[Koonin EV](/wiki/Eugene_Koonin)(July 2011).["Unification of Cas protein families and a simple scenario for the origin and evolution of CRISPR-Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3150331).*Biology Direct*.**6**38.[doi](/wiki/Doi_(identifier)):[10.1186/1745-6150-6-38](https://doi.org/10.1186%2F1745-6150-6-38).[PMC](/wiki/PMC_(identifier))[3150331](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3150331).[PMID](/wiki/PMID_(identifier))[21756346](https://pubmed.ncbi.nlm.nih.gov/21756346). - ^
**a****b****c****d****e****f****g****h****i****j****k****l****m****n****o****p****q****r****s****t****u****v**Makarova KS, Wolf YI, Iranzo J, Shmakov SA, Alkhnbashi OS, Brouns SJ, et al. (February 2020).**w**["Evolutionary classification of CRISPR-Cas systems: a burst of class 2 and derived variants"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8905525).*Nature Reviews. Microbiology*.**18**(2): 67–83.[doi](/wiki/Doi_(identifier)):[10.1038/s41579-019-0299-x](https://doi.org/10.1038%2Fs41579-019-0299-x).[hdl](/wiki/Hdl_(identifier)):[10045/102627](https://hdl.handle.net/10045%2F102627).[PMC](/wiki/PMC_(identifier))[8905525](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8905525).[PMID](/wiki/PMID_(identifier))[31857715](https://pubmed.ncbi.nlm.nih.gov/31857715). - ^
**a**Mogila I, Kazlauskiene M, Valinskyte S, Tamulaitiene G, Tamulaitis G, Siksnys V (March 2019).**b**["Genetic Dissection of the Type III-A CRISPR-Cas System Csm Complex Reveals Roles of Individual Subunits"](https://doi.org/10.1016%2Fj.celrep.2019.02.029).*Cell Reports*.**26**(10): 2753–2765.e4.[doi](/wiki/Doi_(identifier)):[10.1016/j.celrep.2019.02.029](https://doi.org/10.1016%2Fj.celrep.2019.02.029).[PMID](/wiki/PMID_(identifier))[30840895](https://pubmed.ncbi.nlm.nih.gov/30840895). - ^
**a****b**Gasiunas G,**c**[Barrangou R](/wiki/Rodolphe_Barrangou), Horvath P, Siksnys V (September 2012).["Cas9-crRNA ribonucleoprotein complex mediates specific DNA cleavage for adaptive immunity in bacteria"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3465414).*Proceedings of the National Academy of Sciences of the United States of America*.**109**(39): E2579–2586.[Bibcode](/wiki/Bibcode_(identifier)):[2012PNAS..109E2579G](https://ui.adsabs.harvard.edu/abs/2012PNAS..109E2579G).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.1208507109](https://doi.org/10.1073%2Fpnas.1208507109).[PMC](/wiki/PMC_(identifier))[3465414](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3465414).[PMID](/wiki/PMID_(identifier))[22949671](https://pubmed.ncbi.nlm.nih.gov/22949671). Heler R, Samai P, Modell JW, Weiner C, Goldberg GW, Bikard D, et al. (March 2015).[^](#cite_ref-74)["Cas9 specifies functional viral targets during CRISPR-Cas adaptation"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4385744).*Nature*.**519**(7542): 199–202.[Bibcode](/wiki/Bibcode_(identifier)):[2015Natur.519..199H](https://ui.adsabs.harvard.edu/abs/2015Natur.519..199H).[doi](/wiki/Doi_(identifier)):[10.1038/nature14245](https://doi.org/10.1038%2Fnature14245).[PMC](/wiki/PMC_(identifier))[4385744](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4385744).[PMID](/wiki/PMID_(identifier))[25707807](https://pubmed.ncbi.nlm.nih.gov/25707807).Nam KH, Kurinov I, Ke A (September 2011).[^](#cite_ref-75)["Crystal structure of clustered regularly interspaced short palindromic repeats (CRISPR)-associated Csn2 protein revealed Ca2+-dependent double-stranded DNA binding activity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3162437).*The Journal of Biological Chemistry*.**286**(35): 30759–30768.[doi](/wiki/Doi_(identifier)):[10.1074/jbc.M111.256263](https://doi.org/10.1074%2Fjbc.M111.256263).[PMC](/wiki/PMC_(identifier))[3162437](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3162437).[PMID](/wiki/PMID_(identifier))[21697083](https://pubmed.ncbi.nlm.nih.gov/21697083).Lee H, Dhingra Y, Sashital DG (April 2019).[^](#cite_ref-76)["The Cas4-Cas1-Cas2 complex mediates precise prespacer processing during CRISPR adaptation"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6519985).*eLife*.**8**e44248.[doi](/wiki/Doi_(identifier)):[10.7554/eLife.44248](https://doi.org/10.7554%2FeLife.44248).[PMC](/wiki/PMC_(identifier))[6519985](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6519985).[PMID](/wiki/PMID_(identifier))[31021314](https://pubmed.ncbi.nlm.nih.gov/31021314).Chylinski K, Le Rhun A, Charpentier E (May 2013).[^](#cite_ref-pmid23563642_77-0)["The tracrRNA and Cas9 families of type II CRISPR-Cas immunity systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737331).*RNA Biology*.**10**(5): 726–737.[doi](/wiki/Doi_(identifier)):[10.4161/rna.24321](https://doi.org/10.4161%2Frna.24321).[PMC](/wiki/PMC_(identifier))[3737331](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737331).[PMID](/wiki/PMID_(identifier))[23563642](https://pubmed.ncbi.nlm.nih.gov/23563642).Makarova KS, Zhang F, Koonin EV (January 2017).[^](#cite_ref-78)["SnapShot: Class 2 CRISPR-Cas Systems"](https://doi.org/10.1016%2Fj.cell.2016.12.038).*Cell*.**168**(1–2): 328–328.e1.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2016.12.038](https://doi.org/10.1016%2Fj.cell.2016.12.038).[PMID](/wiki/PMID_(identifier))[28086097](https://pubmed.ncbi.nlm.nih.gov/28086097).Paul B, Montoya G (February 2020).[^](#cite_ref-79)["CRISPR-Cas12a: Functional overview and applications"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7090318).*Biomedical Journal*.**43**(1): 8–17.[doi](/wiki/Doi_(identifier)):[10.1016/j.bj.2019.10.005](https://doi.org/10.1016%2Fj.bj.2019.10.005).[PMC](/wiki/PMC_(identifier))[7090318](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7090318).[PMID](/wiki/PMID_(identifier))[32200959](https://pubmed.ncbi.nlm.nih.gov/32200959).Cox DB, Gootenberg JS, Abudayyeh OO, Franklin B, Kellner MJ, Joung J, et al. (November 2017).[^](#cite_ref-81)["RNA editing with CRISPR-Cas13"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5793859).*Science*.**358**(6366): 1019–1027.[Bibcode](/wiki/Bibcode_(identifier)):[2017Sci...358.1019C](https://ui.adsabs.harvard.edu/abs/2017Sci...358.1019C).[doi](/wiki/Doi_(identifier)):[10.1126/science.aaq0180](https://doi.org/10.1126%2Fscience.aaq0180).[PMC](/wiki/PMC_(identifier))[5793859](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5793859).[PMID](/wiki/PMID_(identifier))[29070703](https://pubmed.ncbi.nlm.nih.gov/29070703).- ^
**a**Xu C, Zhou Y, Xiao Q, He B, Geng G, Wang Z, et al. (May 2021).**b**["Programmable RNA editing with compact CRISPR-Cas13 systems from uncultivated microbes"](https://doi.org/10.1038%2Fs41592-021-01124-4).*Nature Methods*.**18**(5): 499–506.[doi](/wiki/Doi_(identifier)):[10.1038/s41592-021-01124-4](https://doi.org/10.1038%2Fs41592-021-01124-4).[PMID](/wiki/PMID_(identifier))[33941935](https://pubmed.ncbi.nlm.nih.gov/33941935). Azangou-Khyavy M, Ghasemi M, Khanali J, Boroomand-Saboor M, Jamalkhah M, Soleimani M, et al. (2020).[^](#cite_ref-pmid33117331_83-0)["CRISPR/Cas: From Tumor Gene Editing to T Cell-Based Immunotherapy of Cancer"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7553049).*Frontiers in Immunology*.**11**2062.[doi](/wiki/Doi_(identifier)):[10.3389/fimmu.2020.02062](https://doi.org/10.3389%2Ffimmu.2020.02062).[PMC](/wiki/PMC_(identifier))[7553049](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7553049).[PMID](/wiki/PMID_(identifier))[33117331](https://pubmed.ncbi.nlm.nih.gov/33117331).- ^
**a**Aliyari R, Ding SW (January 2009).**b**["RNA-based viral immunity initiated by the Dicer family of host immune receptors"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2676720).*Immunological Reviews*.**227**(1): 176–188.[doi](/wiki/Doi_(identifier)):[10.1111/j.1600-065X.2008.00722.x](https://doi.org/10.1111%2Fj.1600-065X.2008.00722.x).[PMC](/wiki/PMC_(identifier))[2676720](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2676720).[PMID](/wiki/PMID_(identifier))[19120484](https://pubmed.ncbi.nlm.nih.gov/19120484). Dugar G, Herbig A, Förstner KU, Heidrich N, Reinhardt R, Nieselt K, et al. (May 2013).[^](#cite_ref-pmid23696746_85-0)["High-resolution transcriptome maps reveal strain-specific regulatory features of multiple Campylobacter jejuni isolates"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3656092).*PLOS Genetics*.**9**(5) e1003495.[doi](/wiki/Doi_(identifier)):[10.1371/journal.pgen.1003495](https://doi.org/10.1371%2Fjournal.pgen.1003495).[PMC](/wiki/PMC_(identifier))[3656092](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3656092).[PMID](/wiki/PMID_(identifier))[23696746](https://pubmed.ncbi.nlm.nih.gov/23696746).Hatoum-Aslan A, Maniv I, Marraffini LA (December 2011).[^](#cite_ref-pmid22160698_86-0)["Mature clustered, regularly interspaced, short palindromic repeats RNA (crRNA) length is measured by a ruler mechanism anchored at the precursor processing site"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3248500).*Proceedings of the National Academy of Sciences of the United States of America*.**108**(52): 21218–21222.[Bibcode](/wiki/Bibcode_(identifier)):[2011PNAS..10821218H](https://ui.adsabs.harvard.edu/abs/2011PNAS..10821218H).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.1112832108](https://doi.org/10.1073%2Fpnas.1112832108).[PMC](/wiki/PMC_(identifier))[3248500](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3248500).[PMID](/wiki/PMID_(identifier))[22160698](https://pubmed.ncbi.nlm.nih.gov/22160698).- ^
**a**Yosef I, Goren MG, Qimron U (July 2012).**b**["Proteins and DNA elements essential for the CRISPR adaptation process in](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3384332).*Escherichia coli*"*Nucleic Acids Research*.**40**(12): 5569–5576.[doi](/wiki/Doi_(identifier)):[10.1093/nar/gks216](https://doi.org/10.1093%2Fnar%2Fgks216).[PMC](/wiki/PMC_(identifier))[3384332](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3384332).[PMID](/wiki/PMID_(identifier))[22402487](https://pubmed.ncbi.nlm.nih.gov/22402487). - ^
**a****b****c**Swarts DC, Mosterd C, van Passel MW, Brouns SJ (2012).**d**["CRISPR interference directs strand specific spacer acquisition"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3338789).*PLOS ONE*.**7**(4) e35888.[Bibcode](/wiki/Bibcode_(identifier)):[2012PLoSO...735888S](https://ui.adsabs.harvard.edu/abs/2012PLoSO...735888S).[doi](/wiki/Doi_(identifier)):[10.1371/journal.pone.0035888](https://doi.org/10.1371%2Fjournal.pone.0035888).[PMC](/wiki/PMC_(identifier))[3338789](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3338789).[PMID](/wiki/PMID_(identifier))[22558257](https://pubmed.ncbi.nlm.nih.gov/22558257). [^](#cite_ref-pmid21219465_89-0)[Babu M](/wiki/Eugene_Koonin), Beloglazova N, Flick R, Graham C, Skarina T, Nocek B, et al. (January 2011).["A dual function of the CRISPR-Cas system in bacterial antivirus immunity and DNA repair"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3071548).*Molecular Microbiology*.**79**(2): 484–502.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.2010.07465.x](https://doi.org/10.1111%2Fj.1365-2958.2010.07465.x).[PMC](/wiki/PMC_(identifier))[3071548](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3071548).[PMID](/wiki/PMID_(identifier))[21219465](https://pubmed.ncbi.nlm.nih.gov/21219465).Han D, Lehmann K, Krauss G (June 2009). "SSO1450—a CAS1 protein from Sulfolobus solfataricus P2 with high affinity for RNA and DNA".[^](#cite_ref-pmid19427858_90-0)*FEBS Letters*.**583**(12): 1928–1932.[Bibcode](/wiki/Bibcode_(identifier)):[2009FEBSL.583.1928H](https://ui.adsabs.harvard.edu/abs/2009FEBSL.583.1928H).[doi](/wiki/Doi_(identifier)):[10.1016/j.febslet.2009.04.047](https://doi.org/10.1016%2Fj.febslet.2009.04.047).[PMID](/wiki/PMID_(identifier))[19427858](https://pubmed.ncbi.nlm.nih.gov/19427858).Wiedenheft B, Zhou K, Jinek M, Coyle SM, Ma W,[^](#cite_ref-pmid19523907_91-0)[Doudna JA](/wiki/Jennifer_Doudna)(June 2009).["Structural basis for DNase activity of a conserved protein implicated in CRISPR-mediated genome defense"](https://doi.org/10.1016%2Fj.str.2009.03.019).*Structure*.**17**(6): 904–912.[doi](/wiki/Doi_(identifier)):[10.1016/j.str.2009.03.019](https://doi.org/10.1016%2Fj.str.2009.03.019).[PMID](/wiki/PMID_(identifier))[19523907](https://pubmed.ncbi.nlm.nih.gov/19523907).Beloglazova N, Brown G, Zimmerman MD, Proudfoot M, Makarova KS, Kudritska M, et al. (July 2008).[^](#cite_ref-pmid18482976_92-0)["A novel family of sequence-specific endoribonucleases associated with the clustered regularly interspaced short palindromic repeats"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2459268).*The Journal of Biological Chemistry*.**283**(29): 20361–20371.[doi](/wiki/Doi_(identifier)):[10.1074/jbc.M803225200](https://doi.org/10.1074%2Fjbc.M803225200).[PMC](/wiki/PMC_(identifier))[2459268](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2459268).[PMID](/wiki/PMID_(identifier))[18482976](https://pubmed.ncbi.nlm.nih.gov/18482976).Samai P, Smith P, Shuman S (December 2010).[^](#cite_ref-pmid21139194_93-0)["Structure of a CRISPR-associated protein Cas2 from Desulfovibrio vulgaris"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2998353).*Acta Crystallographica Section F*.**66**(Pt 12): 1552–1556.[doi](/wiki/Doi_(identifier)):[10.1107/S1744309110039801](https://doi.org/10.1107%2FS1744309110039801).[PMC](/wiki/PMC_(identifier))[2998353](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2998353).[PMID](/wiki/PMID_(identifier))[21139194](https://pubmed.ncbi.nlm.nih.gov/21139194).Nam KH, Ding F, Haitjema C, Huang Q, DeLisa MP, Ke A (October 2012).[^](#cite_ref-pmid22942283_94-0)["Double-stranded endonuclease activity in Bacillus halodurans clustered regularly interspaced short palindromic repeats (CRISPR)-associated Cas2 protein"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3476262).*The Journal of Biological Chemistry*.**287**(43): 35943–35952.[doi](/wiki/Doi_(identifier)):[10.1074/jbc.M112.382598](https://doi.org/10.1074%2Fjbc.M112.382598).[PMC](/wiki/PMC_(identifier))[3476262](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3476262).[PMID](/wiki/PMID_(identifier))[22942283](https://pubmed.ncbi.nlm.nih.gov/22942283).- ^
**a**Nuñez JK, Kranzusch PJ, Noeske J, Wright AV, Davies CW,**b**[Doudna JA](/wiki/Jennifer_Doudna)(June 2014).["Cas1-Cas2 complex formation mediates spacer acquisition during CRISPR-Cas adaptive immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4075942).*Nature Structural & Molecular Biology*.**21**(6): 528–534.[doi](/wiki/Doi_(identifier)):[10.1038/nsmb.2820](https://doi.org/10.1038%2Fnsmb.2820).[PMC](/wiki/PMC_(identifier))[4075942](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4075942).[PMID](/wiki/PMID_(identifier))[24793649](https://pubmed.ncbi.nlm.nih.gov/24793649). Nuñez JK, Lee AS, Engelman A,[^](#cite_ref-pmid25707795_96-0)[Doudna JA](/wiki/Jennifer_Doudna)(March 2015).["Integrase-mediated spacer acquisition during CRISPR-Cas adaptive immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4359072).*Nature*.**519**(7542): 193–198.[Bibcode](/wiki/Bibcode_(identifier)):[2015Natur.519..193N](https://ui.adsabs.harvard.edu/abs/2015Natur.519..193N).[doi](/wiki/Doi_(identifier)):[10.1038/nature14237](https://doi.org/10.1038%2Fnature14237).[PMC](/wiki/PMC_(identifier))[4359072](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4359072).[PMID](/wiki/PMID_(identifier))[25707795](https://pubmed.ncbi.nlm.nih.gov/25707795).Wang J, Li J, Zhao H, Sheng G, Wang M, Yin M, et al. (November 2015).[^](#cite_ref-pmid26478180_97-0)["Structural and Mechanistic Basis of PAM-Dependent Spacer Acquisition in CRISPR-Cas Systems"](https://doi.org/10.1016%2Fj.cell.2015.10.008).*Cell*.**163**(4): 840–853.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2015.10.008](https://doi.org/10.1016%2Fj.cell.2015.10.008).[PMID](/wiki/PMID_(identifier))[26478180](https://pubmed.ncbi.nlm.nih.gov/26478180).Nuñez JK, Harrington LB, Kranzusch PJ, Engelman AN,[^](#cite_ref-pmid26503043_98-0)[Doudna JA](/wiki/Jennifer_Doudna)(November 2015).["Foreign DNA capture during CRISPR-Cas adaptive immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4662619).*Nature*.**527**(7579): 535–538.[Bibcode](/wiki/Bibcode_(identifier)):[2015Natur.527..535N](https://ui.adsabs.harvard.edu/abs/2015Natur.527..535N).[doi](/wiki/Doi_(identifier)):[10.1038/nature15760](https://doi.org/10.1038%2Fnature15760).[PMC](/wiki/PMC_(identifier))[4662619](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4662619).[PMID](/wiki/PMID_(identifier))[26503043](https://pubmed.ncbi.nlm.nih.gov/26503043).Sorek R, Lawrence CM, Wiedenheft B (2013).[^](#cite_ref-99)["CRISPR-mediated adaptive immune systems in bacteria and archaea"](https://doi.org/10.1146%2Fannurev-biochem-072911-172315).*Annual Review of Biochemistry*.**82**(1): 237–266.[doi](/wiki/Doi_(identifier)):[10.1146/annurev-biochem-072911-172315](https://doi.org/10.1146%2Fannurev-biochem-072911-172315).[PMID](/wiki/PMID_(identifier))[23495939](https://pubmed.ncbi.nlm.nih.gov/23495939).Nuñez JK, Bai L, Harrington LB, Hinder TL,[^](#cite_ref-100)[Doudna JA](/wiki/Jennifer_Doudna)(June 2016).["CRISPR Immunological Memory Requires a Host Factor for Specificity"](https://doi.org/10.1016%2Fj.molcel.2016.04.027).*Molecular Cell*.**62**(6): 824–833.[doi](/wiki/Doi_(identifier)):[10.1016/j.molcel.2016.04.027](https://doi.org/10.1016%2Fj.molcel.2016.04.027).[PMID](/wiki/PMID_(identifier))[27211867](https://pubmed.ncbi.nlm.nih.gov/27211867).Fagerlund RD, Wilkinson ME, Klykov O, Barendregt A, Pearce FG, Kieper SN, et al. (June 2017).[^](#cite_ref-101)["Spacer capture and integration by a type I-F Cas1-Cas2–3 CRISPR adaptation complex"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5495228).*Proceedings of the National Academy of Sciences of the United States of America*.**114**(26): E5122 – E5128.[Bibcode](/wiki/Bibcode_(identifier)):[2017PNAS..114E5122F](https://ui.adsabs.harvard.edu/abs/2017PNAS..114E5122F).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.1618421114](https://doi.org/10.1073%2Fpnas.1618421114).[PMC](/wiki/PMC_(identifier))[5495228](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5495228).[PMID](/wiki/PMID_(identifier))[28611213](https://pubmed.ncbi.nlm.nih.gov/28611213).Rollie C, Graham S, Rouillon C, White MF (February 2018).[^](#cite_ref-102)["Prespacer processing and specific integration in a Type I-A CRISPR system"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5815122).*Nucleic Acids Research*.**46**(3): 1007–1020.[doi](/wiki/Doi_(identifier)):[10.1093/nar/gkx1232](https://doi.org/10.1093%2Fnar%2Fgkx1232).[PMC](/wiki/PMC_(identifier))[5815122](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5815122).[PMID](/wiki/PMID_(identifier))[29228332](https://pubmed.ncbi.nlm.nih.gov/29228332).- ^
**a****b****c**Horvath P, Romero DA, Coûté-Monvoisin AC, Richards M, Deveau H, Moineau S, et al. (February 2008).**d**["Diversity, activity, and evolution of CRISPR loci in](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238196).*Streptococcus thermophilus*"*Journal of Bacteriology*.**190**(4): 1401–1412.[doi](/wiki/Doi_(identifier)):[10.1128/JB.01415-07](https://doi.org/10.1128%2FJB.01415-07).[PMC](/wiki/PMC_(identifier))[2238196](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238196).[PMID](/wiki/PMID_(identifier))[18065539](https://pubmed.ncbi.nlm.nih.gov/18065539). - ^
**a****b**Deveau H,**c**[Barrangou R](/wiki/Rodolphe_Barrangou), Garneau JE, Labonté J, Fremaux C, Boyaval P, et al. (February 2008).["Phage response to CRISPR-encoded resistance in](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238228).*Streptococcus thermophilus*"*Journal of Bacteriology*.**190**(4): 1390–1400.[doi](/wiki/Doi_(identifier)):[10.1128/JB.01412-07](https://doi.org/10.1128%2FJB.01412-07).[PMC](/wiki/PMC_(identifier))[2238228](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238228).[PMID](/wiki/PMID_(identifier))[18065545](https://pubmed.ncbi.nlm.nih.gov/18065545). Mojica FJ, Díez-Villaseñor C, García-Martínez J, Almendros C (March 2009).[^](#cite_ref-pmid19246744_105-0)["Short motif sequences determine the targets of the prokaryotic CRISPR defence system"](https://doi.org/10.1099%2Fmic.0.023960-0).*Microbiology*.**155**(Pt 3): 733–740.[doi](/wiki/Doi_(identifier)):[10.1099/mic.0.023960-0](https://doi.org/10.1099%2Fmic.0.023960-0).[PMID](/wiki/PMID_(identifier))[19246744](https://pubmed.ncbi.nlm.nih.gov/19246744).- ^
**a**Lillestøl RK, Shah SA, Brügger K, Redder P, Phan H, Christiansen J, et al. (April 2009). "CRISPR families of the crenarchaeal genus Sulfolobus: bidirectional transcription and dynamic properties".**b***Molecular Microbiology*.**72**(1): 259–272.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.2009.06641.x](https://doi.org/10.1111%2Fj.1365-2958.2009.06641.x).[PMID](/wiki/PMID_(identifier))[19239620](https://pubmed.ncbi.nlm.nih.gov/19239620). - ^
**a**Shah SA, Hansen NR, Garrett RA (February 2009). "Distribution of CRISPR spacer matches in viruses and plasmids of crenarchaeal acidothermophiles and implications for their inhibitory mechanism".**b***Biochemical Society Transactions*.**37**(Pt 1): 23–28.[doi](/wiki/Doi_(identifier)):[10.1042/BST0370023](https://doi.org/10.1042%2FBST0370023).[PMID](/wiki/PMID_(identifier))[19143596](https://pubmed.ncbi.nlm.nih.gov/19143596). - ^
**a**Díez-Villaseñor C, Guzmán NM, Almendros C, García-Martínez J, Mojica FJ (May 2013).**b**["CRISPR-spacer integration reporter plasmids reveal distinct genuine acquisition specificities among CRISPR-Cas I-E variants of](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737337).*Escherichia coli*"*RNA Biology*.**10**(5): 792–802.[doi](/wiki/Doi_(identifier)):[10.4161/rna.24023](https://doi.org/10.4161%2Frna.24023).[PMC](/wiki/PMC_(identifier))[3737337](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737337).[PMID](/wiki/PMID_(identifier))[23445770](https://pubmed.ncbi.nlm.nih.gov/23445770). - ^
**a****b**Erdmann S, Garrett RA (September 2012).**c**["Selective and hyperactive uptake of foreign DNA by adaptive immune systems of an archaeon via two distinct mechanisms"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3468723).*Molecular Microbiology*.**85**(6): 1044–1056.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.2012.08171.x](https://doi.org/10.1111%2Fj.1365-2958.2012.08171.x).[PMC](/wiki/PMC_(identifier))[3468723](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3468723).[PMID](/wiki/PMID_(identifier))[22834906](https://pubmed.ncbi.nlm.nih.gov/22834906). - ^
**a**Shah SA, Erdmann S, Mojica FJ, Garrett RA (May 2013).**b**["Protospacer recognition motifs: mixed identities and functional diversity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737346).*RNA Biology*.**10**(5): 891–899.[doi](/wiki/Doi_(identifier)):[10.4161/rna.23764](https://doi.org/10.4161%2Frna.23764).[PMC](/wiki/PMC_(identifier))[3737346](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737346).[PMID](/wiki/PMID_(identifier))[23403393](https://pubmed.ncbi.nlm.nih.gov/23403393). Andersson AF, Banfield JF (May 2008). "Virus population dynamics and acquired virus resistance in natural microbial communities".[^](#cite_ref-pmid18497291_111-0)*Science*.**320**(5879): 1047–1050.[Bibcode](/wiki/Bibcode_(identifier)):[2008Sci...320.1047A](https://ui.adsabs.harvard.edu/abs/2008Sci...320.1047A).[doi](/wiki/Doi_(identifier)):[10.1126/science.1157358](https://doi.org/10.1126%2Fscience.1157358).[PMID](/wiki/PMID_(identifier))[18497291](https://pubmed.ncbi.nlm.nih.gov/18497291).- ^
**a****b****c**Pride DT, Sun CL, Salzman J, Rao N, Loomer P, Armitage GC, et al. (January 2011).**d**["Analysis of streptococcal CRISPRs from human saliva reveals substantial sequence diversity within and between subjects over time"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3012920).*Genome Research*.**21**(1): 126–136.[doi](/wiki/Doi_(identifier)):[10.1101/gr.111732.110](https://doi.org/10.1101%2Fgr.111732.110).[PMC](/wiki/PMC_(identifier))[3012920](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3012920).[PMID](/wiki/PMID_(identifier))[21149389](https://pubmed.ncbi.nlm.nih.gov/21149389). - ^
**a**Goren MG, Yosef I, Auster O, Qimron U (October 2012). "Experimental definition of a clustered regularly interspaced short palindromic duplicon in**b***Escherichia coli*".*Journal of Molecular Biology*.**423**(1): 14–16.[doi](/wiki/Doi_(identifier)):[10.1016/j.jmb.2012.06.037](https://doi.org/10.1016%2Fj.jmb.2012.06.037).[PMID](/wiki/PMID_(identifier))[22771574](https://pubmed.ncbi.nlm.nih.gov/22771574). - ^
**a****b**Datsenko KA, Pougach K, Tikhonov A, Wanner BL, Severinov K, Semenova E (July 2012).**c**["Molecular memory of prior infections activates the CRISPR/Cas adaptive bacterial immunity system"](https://doi.org/10.1038%2Fncomms1937).*Nature Communications*.**3**945.[Bibcode](/wiki/Bibcode_(identifier)):[2012NatCo...3..945D](https://ui.adsabs.harvard.edu/abs/2012NatCo...3..945D).[doi](/wiki/Doi_(identifier)):[10.1038/ncomms1937](https://doi.org/10.1038%2Fncomms1937).[PMID](/wiki/PMID_(identifier))[22781758](https://pubmed.ncbi.nlm.nih.gov/22781758). Gesner EM, Schellenberg MJ, Garside EL, George MM, Macmillan AM (June 2011). "Recognition and maturation of effector RNAs in a CRISPR interference pathway".[^](#cite_ref-pmid21572444_115-0)*Nature Structural & Molecular Biology*.**18**(6): 688–692.[doi](/wiki/Doi_(identifier)):[10.1038/nsmb.2042](https://doi.org/10.1038%2Fnsmb.2042).[PMID](/wiki/PMID_(identifier))[21572444](https://pubmed.ncbi.nlm.nih.gov/21572444).Sashital DG, Jinek M,[^](#cite_ref-pmid21572442_116-0)[Doudna JA](/wiki/Jennifer_Doudna)(June 2011). "An RNA-induced conformational change required for CRISPR RNA cleavage by the endoribonuclease Cse3".*Nature Structural & Molecular Biology*.**18**(6): 680–687.[doi](/wiki/Doi_(identifier)):[10.1038/nsmb.2043](https://doi.org/10.1038%2Fnsmb.2043).[PMID](/wiki/PMID_(identifier))[21572442](https://pubmed.ncbi.nlm.nih.gov/21572442).Haurwitz RE, Jinek M, Wiedenheft B, Zhou K,[^](#cite_ref-pmid20829488_117-0)[Doudna JA](/wiki/Jennifer_Doudna)(September 2010).["Sequence- and structure-specific RNA processing by a CRISPR endonuclease"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3133607).*Science*.**329**(5997): 1355–1358.[Bibcode](/wiki/Bibcode_(identifier)):[2010Sci...329.1355H](https://ui.adsabs.harvard.edu/abs/2010Sci...329.1355H).[doi](/wiki/Doi_(identifier)):[10.1126/science.1192272](https://doi.org/10.1126%2Fscience.1192272).[PMC](/wiki/PMC_(identifier))[3133607](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3133607).[PMID](/wiki/PMID_(identifier))[20829488](https://pubmed.ncbi.nlm.nih.gov/20829488).- ^
**a**Kunin V, Sorek R, Hugenholtz P (2007).**b**["Evolutionary conservation of sequence and secondary structures in CRISPR repeats"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1896005).*Genome Biology*.**8**(4) R61.[doi](/wiki/Doi_(identifier)):[10.1186/gb-2007-8-4-r61](https://doi.org/10.1186%2Fgb-2007-8-4-r61).[PMC](/wiki/PMC_(identifier))[1896005](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1896005).[PMID](/wiki/PMID_(identifier))[17442114](https://pubmed.ncbi.nlm.nih.gov/17442114). Carte J, Wang R, Li H, Terns RM, Terns MP (December 2008).[^](#cite_ref-pmid19141480_119-0)["Cas6 is an endoribonuclease that generates guide RNAs for invader defense in prokaryotes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2607076).*Genes & Development*.**22**(24): 3489–3496.[doi](/wiki/Doi_(identifier)):[10.1101/gad.1742908](https://doi.org/10.1101%2Fgad.1742908).[PMC](/wiki/PMC_(identifier))[2607076](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2607076).[PMID](/wiki/PMID_(identifier))[19141480](https://pubmed.ncbi.nlm.nih.gov/19141480).Wang R, Preamplume G, Terns MP, Terns RM, Li H (February 2011).[^](#cite_ref-pmid21300293_120-0)["Interaction of the Cas6 riboendonuclease with CRISPR RNAs: recognition and cleavage"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3154685).*Structure*.**19**(2): 257–264.[doi](/wiki/Doi_(identifier)):[10.1016/j.str.2010.11.014](https://doi.org/10.1016%2Fj.str.2010.11.014).[PMC](/wiki/PMC_(identifier))[3154685](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3154685).[PMID](/wiki/PMID_(identifier))[21300293](https://pubmed.ncbi.nlm.nih.gov/21300293).Niewoehner O, Jinek M,[^](#cite_ref-pmid24150936_121-0)[Doudna JA](/wiki/Jennifer_Doudna)(January 2014).["Evolution of CRISPR RNA recognition and processing by Cas6 endonucleases"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3902920).*Nucleic Acids Research*.**42**(2): 1341–1353.[doi](/wiki/Doi_(identifier)):[10.1093/nar/gkt922](https://doi.org/10.1093%2Fnar%2Fgkt922).[PMC](/wiki/PMC_(identifier))[3902920](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3902920).[PMID](/wiki/PMID_(identifier))[24150936](https://pubmed.ncbi.nlm.nih.gov/24150936).Semenova E, Jore MM, Datsenko KA, Semenova A, Westra ER, Wanner B, et al. (June 2011).[^](#cite_ref-pmid21646539_122-0)["Interference by clustered regularly interspaced short palindromic repeat (CRISPR) RNA is governed by a seed sequence"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3121866).*Proceedings of the National Academy of Sciences of the United States of America*.**108**(25): 10098–10103.[Bibcode](/wiki/Bibcode_(identifier)):[2011PNAS..10810098S](https://ui.adsabs.harvard.edu/abs/2011PNAS..10810098S).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.1104144108](https://doi.org/10.1073%2Fpnas.1104144108).[PMC](/wiki/PMC_(identifier))[3121866](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3121866).[PMID](/wiki/PMID_(identifier))[21646539](https://pubmed.ncbi.nlm.nih.gov/21646539).Gudbergsdottir S, Deng L, Chen Z, Jensen JV, Jensen LR, She Q, et al. (January 2011).[^](#cite_ref-pmid21166892_123-0)["Dynamic properties of the Sulfolobus CRISPR/Cas and CRISPR/Cmr systems when challenged with vector-borne viral and plasmid genes and protospacers"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3025118).*Molecular Microbiology*.**79**(1): 35–49.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.2010.07452.x](https://doi.org/10.1111%2Fj.1365-2958.2010.07452.x).[PMC](/wiki/PMC_(identifier))[3025118](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3025118).[PMID](/wiki/PMID_(identifier))[21166892](https://pubmed.ncbi.nlm.nih.gov/21166892).Manica A, Zebec Z, Teichmann D, Schleper C (April 2011).[^](#cite_ref-pmid21385233_124-0)["In vivo activity of CRISPR-mediated virus defence in a hyperthermophilic archaeon"](https://doi.org/10.1111%2Fj.1365-2958.2011.07586.x).*Molecular Microbiology*.**80**(2): 481–491.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.2011.07586.x](https://doi.org/10.1111%2Fj.1365-2958.2011.07586.x).[PMID](/wiki/PMID_(identifier))[21385233](https://pubmed.ncbi.nlm.nih.gov/21385233).Jore MM, Lundgren M, van Duijn E, Bultema JB, Westra ER, Waghmare SP, et al. (May 2011).[^](#cite_ref-pmid21460843_125-0)["Structural basis for CRISPR RNA-guided DNA recognition by Cascade"](https://pure.rug.nl/ws/files/6761944/2011NatStructMolBiolJore.pdf)(PDF).*Nature Structural & Molecular Biology*.**18**(5): 529–536.[doi](/wiki/Doi_(identifier)):[10.1038/nsmb.2019](https://doi.org/10.1038%2Fnsmb.2019).[PMID](/wiki/PMID_(identifier))[21460843](https://pubmed.ncbi.nlm.nih.gov/21460843).Wiedenheft B, Lander GC, Zhou K, Jore MM, Brouns SJ, van der Oost J, et al. (September 2011).[^](#cite_ref-pmid21938068_126-0)["Structures of the RNA-guided surveillance complex from a bacterial immune system"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4165517).*Nature*.**477**(7365): 486–489.[Bibcode](/wiki/Bibcode_(identifier)):[2011Natur.477..486W](https://ui.adsabs.harvard.edu/abs/2011Natur.477..486W).[doi](/wiki/Doi_(identifier)):[10.1038/nature10402](https://doi.org/10.1038%2Fnature10402).[PMC](/wiki/PMC_(identifier))[4165517](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4165517).[PMID](/wiki/PMID_(identifier))[21938068](https://pubmed.ncbi.nlm.nih.gov/21938068).Zhang J, Rouillon C, Kerou M, Reeks J, Brugger K, Graham S, et al. (February 2012).[^](#cite_ref-pmid22227115_127-0)["Structure and mechanism of the CMR complex for CRISPR-mediated antiviral immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3381847).*Molecular Cell*.**45**(3): 303–313.[doi](/wiki/Doi_(identifier)):[10.1016/j.molcel.2011.12.013](https://doi.org/10.1016%2Fj.molcel.2011.12.013).[PMC](/wiki/PMC_(identifier))[3381847](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3381847).[PMID](/wiki/PMID_(identifier))[22227115](https://pubmed.ncbi.nlm.nih.gov/22227115).- ^
**a**Hale CR, Zhao P, Olson S, Duff MO, Graveley BR, Wells L, et al. (November 2009).**b**["RNA-guided RNA cleavage by a CRISPR RNA-Cas protein complex"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2951265).*Cell*.**139**(5): 945–956.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2009.07.040](https://doi.org/10.1016%2Fj.cell.2009.07.040).[PMC](/wiki/PMC_(identifier))[2951265](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2951265).[PMID](/wiki/PMID_(identifier))[19945378](https://pubmed.ncbi.nlm.nih.gov/19945378). Estrella MA, Kuo FT, Bailey S (2016).[^](#cite_ref-Estrella_2016_129-0)["RNA-activated DNA cleavage by the Type III-B CRISPR–Cas effector complex"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4762430)..[Genes & Development](/wiki/Genes_%26_Development)**30**(4): 460–470.[doi](/wiki/Doi_(identifier)):[10.1101/gad.273722.115](https://doi.org/10.1101%2Fgad.273722.115).[PMC](/wiki/PMC_(identifier))[4762430](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4762430).[PMID](/wiki/PMID_(identifier))[26848046](https://pubmed.ncbi.nlm.nih.gov/26848046).Samai P,[^](#cite_ref-Samai_2015_130-0)[Pyenson N](/wiki/Nicholas_Pyenson), Jiang W, Goldberg GW, Hatoum-Aslan A, Marraffini LA (2015).["Co-transcriptional DNA and RNA Cleavage during Type III CRISPR-Cas Immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4594840).*Cell*.**161**(5): 1164–1174.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2015.04.027](https://doi.org/10.1016%2Fj.cell.2015.04.027).[PMC](/wiki/PMC_(identifier))[4594840](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4594840).[PMID](/wiki/PMID_(identifier))[25959775](https://pubmed.ncbi.nlm.nih.gov/25959775).- ^
**a****b**Marraffini LA, Sontheimer EJ (January 2010).**c**["Self versus non-self discrimination during CRISPR RNA-directed immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2813891).*Nature*.**463**(7280): 568–571.[Bibcode](/wiki/Bibcode_(identifier)):[2010Natur.463..568M](https://ui.adsabs.harvard.edu/abs/2010Natur.463..568M).[doi](/wiki/Doi_(identifier)):[10.1038/nature08703](https://doi.org/10.1038%2Fnature08703).[PMC](/wiki/PMC_(identifier))[2813891](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2813891).[PMID](/wiki/PMID_(identifier))[20072129](https://pubmed.ncbi.nlm.nih.gov/20072129). Krupovic M, Béguin P,[^](#cite_ref-132)[Koonin EV](/wiki/Eugene_Koonin)(August 2017).["Casposons: mobile genetic elements that gave rise to the CRISPR-Cas adaptation machinery"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5665730).*Current Opinion in Microbiology*.**38**: 36–43.[doi](/wiki/Doi_(identifier)):[10.1016/j.mib.2017.04.004](https://doi.org/10.1016%2Fj.mib.2017.04.004).[PMC](/wiki/PMC_(identifier))[5665730](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5665730).[PMID](/wiki/PMID_(identifier))[28472712](https://pubmed.ncbi.nlm.nih.gov/28472712).[^](#cite_ref-133)[Koonin EV](/wiki/Eugene_Koonin), Makarova KS (May 2013).["CRISPR-Cas: evolution of an RNA-based adaptive immunity system in prokaryotes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737325).*RNA Biology*.**10**(5): 679–686.[doi](/wiki/Doi_(identifier)):[10.4161/rna.24022](https://doi.org/10.4161%2Frna.24022).[PMC](/wiki/PMC_(identifier))[3737325](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737325).[PMID](/wiki/PMID_(identifier))[23439366](https://pubmed.ncbi.nlm.nih.gov/23439366).[^](#cite_ref-134)[Koonin EV](/wiki/Eugene_Koonin), Makarova KS,[Zhang F](/wiki/Feng_Zhang)(June 2017).["Diversity, classification and evolution of CRISPR-Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5776717).*Current Opinion in Microbiology*.**37**: 67–78.[doi](/wiki/Doi_(identifier)):[10.1016/j.mib.2017.05.008](https://doi.org/10.1016%2Fj.mib.2017.05.008).[PMC](/wiki/PMC_(identifier))[5776717](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5776717).[PMID](/wiki/PMID_(identifier))[28605718](https://pubmed.ncbi.nlm.nih.gov/28605718).- ^
**a**Shmakov S, Smargon A, Scott D, Cox D, Pyzocha N, Yan W, et al. (March 2017).**b**["Diversity and evolution of class 2 CRISPR-Cas systems"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5851899).*Nature Reviews. Microbiology*.**15**(3): 169–182.[doi](/wiki/Doi_(identifier)):[10.1038/nrmicro.2016.184](https://doi.org/10.1038%2Fnrmicro.2016.184).[PMC](/wiki/PMC_(identifier))[5851899](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5851899).[PMID](/wiki/PMID_(identifier))[28111461](https://pubmed.ncbi.nlm.nih.gov/28111461). Kupczok A, Bollback JP (February 2013).[^](#cite_ref-136)["Probabilistic models for CRISPR spacer content evolution"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3704272).*BMC Evolutionary Biology*.**13**(1) 54.[Bibcode](/wiki/Bibcode_(identifier)):[2013BMCEE..13...54K](https://ui.adsabs.harvard.edu/abs/2013BMCEE..13...54K).[doi](/wiki/Doi_(identifier)):[10.1186/1471-2148-13-54](https://doi.org/10.1186%2F1471-2148-13-54).[PMC](/wiki/PMC_(identifier))[3704272](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3704272).[PMID](/wiki/PMID_(identifier))[23442002](https://pubmed.ncbi.nlm.nih.gov/23442002).Sternberg SH, Richter H, Charpentier E, Qimron U (March 2016). "Adaptation in CRISPR-Cas Systems".[^](#cite_ref-137)*Molecular Cell*.**61**(6): 797–808.[doi](/wiki/Doi_(identifier)):[10.1016/j.molcel.2016.01.030](https://doi.org/10.1016%2Fj.molcel.2016.01.030).[hdl](/wiki/Hdl_(identifier)):[21.11116/0000-0003-E74E-2](https://hdl.handle.net/21.11116%2F0000-0003-E74E-2).[PMID](/wiki/PMID_(identifier))[26949040](https://pubmed.ncbi.nlm.nih.gov/26949040).[^](#cite_ref-pmid19906303_138-0)[Koonin EV](/wiki/Eugene_Koonin), Wolf YI (November 2009).["Is evolution Darwinian or/and Lamarckian?"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2781790).*Biology Direct*.**4**: 42.[doi](/wiki/Doi_(identifier)):[10.1186/1745-6150-4-42](https://doi.org/10.1186%2F1745-6150-4-42).[PMC](/wiki/PMC_(identifier))[2781790](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2781790).[PMID](/wiki/PMID_(identifier))[19906303](https://pubmed.ncbi.nlm.nih.gov/19906303).Weiss A (October 2015).[^](#cite_ref-139)["Lamarckian Illusions"](https://doi.org/10.1016%2Fj.tree.2015.08.003).*Trends in Ecology & Evolution*.**30**(10): 566–568.[Bibcode](/wiki/Bibcode_(identifier)):[2015TEcoE..30..566W](https://ui.adsabs.harvard.edu/abs/2015TEcoE..30..566W).[doi](/wiki/Doi_(identifier)):[10.1016/j.tree.2015.08.003](https://doi.org/10.1016%2Fj.tree.2015.08.003).[PMID](/wiki/PMID_(identifier))[26411613](https://pubmed.ncbi.nlm.nih.gov/26411613).- ^
**a****b**[Koonin EV](/wiki/Eugene_Koonin), Wolf YI (February 2016).["Just how Lamarckian is CRISPR-Cas immunity: the continuum of evolvability mechanisms"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4765028).*Biology Direct*.**11**(1) 9.[doi](/wiki/Doi_(identifier)):[10.1186/s13062-016-0111-z](https://doi.org/10.1186%2Fs13062-016-0111-z).[PMC](/wiki/PMC_(identifier))[4765028](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4765028).[PMID](/wiki/PMID_(identifier))[26912144](https://pubmed.ncbi.nlm.nih.gov/26912144). Heidelberg JF, Nelson WC, Schoenfeld T, Bhaya D (2009). Ahmed N (ed.).[^](#cite_ref-141)["Germ warfare in a microbial mat community: CRISPRs provide insights into the co-evolution of host and viral genomes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2612747).*PLOS ONE*.**4**(1) e4169.[Bibcode](/wiki/Bibcode_(identifier)):[2009PLoSO...4.4169H](https://ui.adsabs.harvard.edu/abs/2009PLoSO...4.4169H).[doi](/wiki/Doi_(identifier)):[10.1371/journal.pone.0004169](https://doi.org/10.1371%2Fjournal.pone.0004169).[PMC](/wiki/PMC_(identifier))[2612747](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2612747).[PMID](/wiki/PMID_(identifier))[19132092](https://pubmed.ncbi.nlm.nih.gov/19132092).Touchon M, Rocha EP (June 2010). Randau L (ed.).[^](#cite_ref-pmid20559554_142-0)["The small, slow and specialized CRISPR and anti-CRISPR of Escherichia and Salmonella"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2886076).*PLOS ONE*.**5**(6) e11126.[Bibcode](/wiki/Bibcode_(identifier)):[2010PLoSO...511126T](https://ui.adsabs.harvard.edu/abs/2010PLoSO...511126T).[doi](/wiki/Doi_(identifier)):[10.1371/journal.pone.0011126](https://doi.org/10.1371%2Fjournal.pone.0011126).[PMC](/wiki/PMC_(identifier))[2886076](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2886076).[PMID](/wiki/PMID_(identifier))[20559554](https://pubmed.ncbi.nlm.nih.gov/20559554).- ^
**a****b**Rho M, Wu YW, Tang H, Doak TG, Ye Y (2012).**c**["Diverse CRISPRs evolving in human microbiomes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3374615).*PLOS Genetics*.**8**(6) e1002441.[doi](/wiki/Doi_(identifier)):[10.1371/journal.pgen.1002441](https://doi.org/10.1371%2Fjournal.pgen.1002441).[PMC](/wiki/PMC_(identifier))[3374615](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3374615).[PMID](/wiki/PMID_(identifier))[22719260](https://pubmed.ncbi.nlm.nih.gov/22719260). - ^
**a**Sun CL,**b**[Barrangou R](/wiki/Rodolphe_Barrangou), Thomas BC, Horvath P, Fremaux C, Banfield JF (February 2013). "Phage mutations in response to CRISPR diversification in a bacterial population".*Environmental Microbiology*.**15**(2): 463–470.[Bibcode](/wiki/Bibcode_(identifier)):[2013EnvMi..15..463S](https://ui.adsabs.harvard.edu/abs/2013EnvMi..15..463S).[doi](/wiki/Doi_(identifier)):[10.1111/j.1462-2920.2012.02879.x](https://doi.org/10.1111%2Fj.1462-2920.2012.02879.x).[PMID](/wiki/PMID_(identifier))[23057534](https://pubmed.ncbi.nlm.nih.gov/23057534). Kuno S, Sako Y, Yoshida T (May 2014).[^](#cite_ref-pmid24586036_145-0)["Diversification of CRISPR within coexisting genotypes in a natural population of the bloom-forming cyanobacterium Microcystis aeruginosa"](https://doi.org/10.1099%2Fmic.0.073494-0).*Microbiology*.**160**(Pt 5): 903–916.[doi](/wiki/Doi_(identifier)):[10.1099/mic.0.073494-0](https://doi.org/10.1099%2Fmic.0.073494-0).[PMID](/wiki/PMID_(identifier))[24586036](https://pubmed.ncbi.nlm.nih.gov/24586036).Sorek R, Kunin V, Hugenholtz P (March 2008).[^](#cite_ref-146)["CRISPR—a widespread system that provides acquired resistance against phages in bacteria and archaea"](https://digital.library.unt.edu/ark:/67531/metadc893045/).*Nature Reviews. Microbiology*.**6**(3): 181–186.[doi](/wiki/Doi_(identifier)):[10.1038/nrmicro1793](https://doi.org/10.1038%2Fnrmicro1793).[OSTI](/wiki/OSTI_(identifier))[926489](https://www.osti.gov/biblio/926489).[PMID](/wiki/PMID_(identifier))[18157154](https://pubmed.ncbi.nlm.nih.gov/18157154).Table 1: Web resources for CRISPR analysis

Pride DT, Salzman J, Relman DA (September 2012).[^](#cite_ref-pmid22583485_147-0)["Comparisons of clustered regularly interspaced short palindromic repeats and viromes in human saliva reveal bacterial adaptations to salivary viruses"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3424356).*Environmental Microbiology*.**14**(9): 2564–2576.[Bibcode](/wiki/Bibcode_(identifier)):[2012EnvMi..14.2564P](https://ui.adsabs.harvard.edu/abs/2012EnvMi..14.2564P).[doi](/wiki/Doi_(identifier)):[10.1111/j.1462-2920.2012.02775.x](https://doi.org/10.1111%2Fj.1462-2920.2012.02775.x).[PMC](/wiki/PMC_(identifier))[3424356](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3424356).[PMID](/wiki/PMID_(identifier))[22583485](https://pubmed.ncbi.nlm.nih.gov/22583485).Held NL, Herrera A, Whitaker RJ (November 2013). "Reassortment of CRISPR repeat-spacer loci in Sulfolobus islandicus".[^](#cite_ref-pmid23701169_148-0)*Environmental Microbiology*.**15**(11): 3065–3076.[Bibcode](/wiki/Bibcode_(identifier)):[2013EnvMi..15.3065H](https://ui.adsabs.harvard.edu/abs/2013EnvMi..15.3065H).[doi](/wiki/Doi_(identifier)):[10.1111/1462-2920.12146](https://doi.org/10.1111%2F1462-2920.12146).[PMID](/wiki/PMID_(identifier))[23701169](https://pubmed.ncbi.nlm.nih.gov/23701169).Held NL, Herrera A, Cadillo-Quiroz H, Whitaker RJ (September 2010).[^](#cite_ref-pmid20927396_149-0)["CRISPR associated diversity within a population of Sulfolobus islandicus"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2946923).*PLOS ONE*.**5**(9) e12988.[Bibcode](/wiki/Bibcode_(identifier)):[2010PLoSO...512988H](https://ui.adsabs.harvard.edu/abs/2010PLoSO...512988H).[doi](/wiki/Doi_(identifier)):[10.1371/journal.pone.0012988](https://doi.org/10.1371%2Fjournal.pone.0012988).[PMC](/wiki/PMC_(identifier))[2946923](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2946923).[PMID](/wiki/PMID_(identifier))[20927396](https://pubmed.ncbi.nlm.nih.gov/20927396).- ^
**a****b**Medvedeva S, Liu Y, Koonin EV, Severinov K, Prangishvili D, Krupovic M (November 2019).**c**["Virus-borne mini-CRISPR arrays are involved in interviral conflicts"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6858448).*Nature Communications*.**10**(1) 5204.[Bibcode](/wiki/Bibcode_(identifier)):[2019NatCo..10.5204M](https://ui.adsabs.harvard.edu/abs/2019NatCo..10.5204M).[doi](/wiki/Doi_(identifier)):[10.1038/s41467-019-13205-2](https://doi.org/10.1038%2Fs41467-019-13205-2).[PMC](/wiki/PMC_(identifier))[6858448](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6858448).[PMID](/wiki/PMID_(identifier))[31729390](https://pubmed.ncbi.nlm.nih.gov/31729390). Skennerton CT, Imelfort M, Tyson GW (May 2013).[^](#cite_ref-pmid23511966_151-0)["Crass: identification and reconstruction of CRISPR from unassembled metagenomic data"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3664793).*Nucleic Acids Research*.**41**(10): e105.[doi](/wiki/Doi_(identifier)):[10.1093/nar/gkt183](https://doi.org/10.1093%2Fnar%2Fgkt183).[PMC](/wiki/PMC_(identifier))[3664793](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3664793).[PMID](/wiki/PMID_(identifier))[23511966](https://pubmed.ncbi.nlm.nih.gov/23511966).Stern A, Mick E, Tirosh I, Sagy O, Sorek R (October 2012).[^](#cite_ref-pmid22732228_152-0)["CRISPR targeting reveals a reservoir of common phages associated with the human gut microbiome"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3460193).*Genome Research*.**22**(10): 1985–1994.[doi](/wiki/Doi_(identifier)):[10.1101/gr.138297.112](https://doi.org/10.1101%2Fgr.138297.112).[PMC](/wiki/PMC_(identifier))[3460193](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3460193).[PMID](/wiki/PMID_(identifier))[22732228](https://pubmed.ncbi.nlm.nih.gov/22732228).Novick RP, Christie GE, Penadés JR (August 2010).[^](#cite_ref-pmid20634809_153-0)["The phage-related chromosomal islands of Gram-positive bacteria"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3522866).*Nature Reviews Microbiology*.**8**(8): 541–551.[doi](/wiki/Doi_(identifier)):[10.1038/nrmicro2393](https://doi.org/10.1038%2Fnrmicro2393).[PMC](/wiki/PMC_(identifier))[3522866](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3522866).[PMID](/wiki/PMID_(identifier))[20634809](https://pubmed.ncbi.nlm.nih.gov/20634809).Ram G, Chen J, Kumar K, Ross HF, Ubeda C, Damle PK, et al. (October 2012).[^](#cite_ref-154)["Staphylococcal pathogenicity island interference with helper phage reproduction is a paradigm of molecular parasitism"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3479557).*Proceedings of the National Academy of Sciences of the United States of America*.**109**(40): 16300–16305.[Bibcode](/wiki/Bibcode_(identifier)):[2012PNAS..10916300R](https://ui.adsabs.harvard.edu/abs/2012PNAS..10916300R).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.1204615109](https://doi.org/10.1073%2Fpnas.1204615109).[PMC](/wiki/PMC_(identifier))[3479557](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3479557).[PMID](/wiki/PMID_(identifier))[22991467](https://pubmed.ncbi.nlm.nih.gov/22991467).Ram G, Chen J, Ross HF, Novick RP (October 2014).[^](#cite_ref-155)["Precisely modulated pathogenicity island interference with late phage gene transcription"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4209980).*Proceedings of the National Academy of Sciences of the United States of America*.**111**(40): 14536–14541.[Bibcode](/wiki/Bibcode_(identifier)):[2014PNAS..11114536R](https://ui.adsabs.harvard.edu/abs/2014PNAS..11114536R).[doi](/wiki/Doi_(identifier)):[10.1073/pnas.1406749111](https://doi.org/10.1073%2Fpnas.1406749111).[PMC](/wiki/PMC_(identifier))[4209980](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4209980).[PMID](/wiki/PMID_(identifier))[25246539](https://pubmed.ncbi.nlm.nih.gov/25246539).Seed KD, Lazinski DW, Calderwood SB, Camilli A (February 2013).[^](#cite_ref-pmid23446421_156-0)["A bacteriophage encodes its own CRISPR/Cas adaptive response to evade host innate immunity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3587790).*Nature*.**494**(7438): 489–491.[Bibcode](/wiki/Bibcode_(identifier)):[2013Natur.494..489S](https://ui.adsabs.harvard.edu/abs/2013Natur.494..489S).[doi](/wiki/Doi_(identifier)):[10.1038/nature11927](https://doi.org/10.1038%2Fnature11927).[PMC](/wiki/PMC_(identifier))[3587790](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3587790).[PMID](/wiki/PMID_(identifier))[23446421](https://pubmed.ncbi.nlm.nih.gov/23446421).Boyd CM, Angermeyer A, Hays SG, Barth ZK, Patel KM, Seed KD (September 2021).[^](#cite_ref-Boyd_157-0)["Bacteriophage ICP1: A Persistent Predator of](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9040626).*Vibrio cholerae*"*Annual Review of Virology*.**8**(1): 285–304.[doi](/wiki/Doi_(identifier)):[10.1146/annurev-virology-091919-072020](https://doi.org/10.1146%2Fannurev-virology-091919-072020).[ISSN](/wiki/ISSN_(identifier))[2327-056X](https://search.worldcat.org/issn/2327-056X).[PMC](/wiki/PMC_(identifier))[9040626](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9040626).[PMID](/wiki/PMID_(identifier))[34314595](https://pubmed.ncbi.nlm.nih.gov/34314595).- ^
**a**Koblan LW, Liu DR, Anzalone AV (July 2020).**b**["Genome editing with CRISPR–Cas nucleases, base editors, transposases and prime editors"](https://www.nature.com/articles/s41587-020-0561-9).*Nature Biotechnology*.**38**(7): 824–844.[doi](/wiki/Doi_(identifier)):[10.1038/s41587-020-0561-9](https://doi.org/10.1038%2Fs41587-020-0561-9).[ISSN](/wiki/ISSN_(identifier))[1546-1696](https://search.worldcat.org/issn/1546-1696).[PMID](/wiki/PMID_(identifier))[32572269](https://pubmed.ncbi.nlm.nih.gov/32572269). Randolph PB, Davis JR, Sousa AA, Koblan LW, Levy JM, Chen PJ, et al. (December 2019).[^](#cite_ref-159)["Search-and-replace genome editing without double-strand breaks or donor DNA"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6907074).*Nature*.**576**(7785): 149–157.[Bibcode](/wiki/Bibcode_(identifier)):[2019Natur.576..149A](https://ui.adsabs.harvard.edu/abs/2019Natur.576..149A).[doi](/wiki/Doi_(identifier)):[10.1038/s41586-019-1711-4](https://doi.org/10.1038%2Fs41586-019-1711-4).[ISSN](/wiki/ISSN_(identifier))[1476-4687](https://search.worldcat.org/issn/1476-4687).[PMC](/wiki/PMC_(identifier))[6907074](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6907074).[PMID](/wiki/PMID_(identifier))[31634902](https://pubmed.ncbi.nlm.nih.gov/31634902).Feng Q, Li Q, Zhou H, Wang Z, Lin C, Jiang Z, et al. (August 2024).[^](#cite_ref-Feng_2024_160-0)["CRISPR technology in human diseases"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11286548).*MedComm*.**5**(8) e672.[doi](/wiki/Doi_(identifier)):[10.1002/mco2.672](https://doi.org/10.1002%2Fmco2.672).[PMC](/wiki/PMC_(identifier))[11286548](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11286548).[PMID](/wiki/PMID_(identifier))[39081515](https://pubmed.ncbi.nlm.nih.gov/39081515).Li T, Li S, Kang Y, Zhou J, Yi M (August 2024).[^](#cite_ref-Li_2024_161-0)["Harnessing the evolving CRISPR/Cas9 for precision oncology"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11312220).*Journal of Translational Medicine*.**22**(1) 749.[doi](/wiki/Doi_(identifier)):[10.1186/s12967-024-05570-4](https://doi.org/10.1186%2Fs12967-024-05570-4).[PMC](/wiki/PMC_(identifier))[11312220](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11312220).[PMID](/wiki/PMID_(identifier))[39118151](https://pubmed.ncbi.nlm.nih.gov/39118151).Kaur R, Gupta S, Chauhan A, Mishra V, Sharma MK, Singh J (August 2024). "Harnessing the power of clustered regularly interspaced short palindromic repeats (CRISPR) based microfluidics for next-generation molecular diagnostics".[^](#cite_ref-Kaur_2024_162-0)*Molecular Biology Reports*.**51**(1) 896.[doi](/wiki/Doi_(identifier)):[10.1007/s11033-024-09840-8](https://doi.org/10.1007%2Fs11033-024-09840-8).[PMID](/wiki/PMID_(identifier))[39115550](https://pubmed.ncbi.nlm.nih.gov/39115550).Mishra S, Nayak S, Tuteja N, Poosapati S, Swain DM, Sahoo RK (July 2024).[^](#cite_ref-Mishra_2024_163-0)["CRISPR/Cas-Mediated Genome Engineering in Plants: Application and Prospectives"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11279650).*Plants*.**13**(14). Basel, Switzerland: 1884.[Bibcode](/wiki/Bibcode_(identifier)):[2024Plnts..13.1884M](https://ui.adsabs.harvard.edu/abs/2024Plnts..13.1884M).[doi](/wiki/Doi_(identifier)):[10.3390/plants13141884](https://doi.org/10.3390%2Fplants13141884).[PMC](/wiki/PMC_(identifier))[11279650](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11279650).[PMID](/wiki/PMID_(identifier))[39065411](https://pubmed.ncbi.nlm.nih.gov/39065411).Shinwari ZK, Tanveer F, Khalil AT (2018).[^](#cite_ref-Shinwari_2018_164-0)["Ethical Issues Regarding CRISPR Mediated Genome Editing"](https://doi.org/10.21775%2Fcimb.026.103).*Current Issues in Molecular Biology*.**26**: 103–110.[doi](/wiki/Doi_(identifier)):[10.21775/cimb.026.103](https://doi.org/10.21775%2Fcimb.026.103).[PMID](/wiki/PMID_(identifier))[28879860](https://pubmed.ncbi.nlm.nih.gov/28879860).Kozan DW, Farber SA (February 2024).[^](#cite_ref-Kozan_2024_165-0)["Is It Ever Wise to Edit Wild-Type Alleles? Engineered CRISPR Alleles Versus Millions of Years of Human Evolution"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10948015).*Arteriosclerosis, Thrombosis, and Vascular Biology*.**44**(2): 328–333.[doi](/wiki/Doi_(identifier)):[10.1161/ATVBAHA.123.318069](https://doi.org/10.1161%2FATVBAHA.123.318069).[PMC](/wiki/PMC_(identifier))[10948015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10948015).[PMID](/wiki/PMID_(identifier))[38059350](https://pubmed.ncbi.nlm.nih.gov/38059350).

## Further reading

[[edit](/w/index.php?title=CRISPR&action=edit§ion=26)]

[Doudna J](/wiki/Jennifer_Doudna), Mali P (23 March 2016).*CRISPR-Cas: A Laboratory Manual*. New York: Cold Spring Harbor Laboratory Press.[ISBN](/wiki/ISBN_(identifier))[978-1-62182-131-1](/wiki/Special:BookSources/978-1-62182-131-1).- Mohanraju P, Makarova KS, Zetsche B,
[Zhang F](/wiki/Feng_Zhang),[Koonin EV](/wiki/Eugene_Koonin), van der Oost J (August 2016). "Diverse evolutionary roots and mechanistic variations of the CRISPR-Cas systems".*Science*.**353**(6299) aad5147.[doi](/wiki/Doi_(identifier)):[10.1126/science.aad5147](https://doi.org/10.1126%2Fscience.aad5147).[hdl](/wiki/Hdl_(identifier)):[1721.1/113195](https://hdl.handle.net/1721.1%2F113195).[PMID](/wiki/PMID_(identifier))[27493190](https://pubmed.ncbi.nlm.nih.gov/27493190). - Sander JD, Joung JK (April 2014).
["CRISPR-Cas systems for editing, regulating and targeting genomes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4022601).*Nature Biotechnology*.**32**(4): 347–355.[doi](/wiki/Doi_(identifier)):[10.1038/nbt.2842](https://doi.org/10.1038%2Fnbt.2842).[PMC](/wiki/PMC_(identifier))[4022601](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4022601).[PMID](/wiki/PMID_(identifier))[24584096](https://pubmed.ncbi.nlm.nih.gov/24584096). - Slaymaker IM, Gao L, Zetsche B, Scott DA, Yan WX,
[Zhang F](/wiki/Feng_Zhang)(January 2016).["Rationally engineered Cas9 nucleases with improved specificity"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4714946).*Science*.**351**(6268): 84–88.[Bibcode](/wiki/Bibcode_(identifier)):[2016Sci...351...84S](https://ui.adsabs.harvard.edu/abs/2016Sci...351...84S).[doi](/wiki/Doi_(identifier)):[10.1126/science.aad5227](https://doi.org/10.1126%2Fscience.aad5227).[PMC](/wiki/PMC_(identifier))[4714946](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4714946).[PMID](/wiki/PMID_(identifier))[26628643](https://pubmed.ncbi.nlm.nih.gov/26628643). - Terns RM, Terns MP (March 2014).
["CRISPR-based technologies: prokaryotic defense weapons repurposed"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3981743).*Trends in Genetics*.**30**(3): 111–118.[doi](/wiki/Doi_(identifier)):[10.1016/j.tig.2014.01.003](https://doi.org/10.1016%2Fj.tig.2014.01.003).[PMC](/wiki/PMC_(identifier))[3981743](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3981743).[PMID](/wiki/PMID_(identifier))[24555991](https://pubmed.ncbi.nlm.nih.gov/24555991). - Westra ER, Buckling A, Fineran PC (May 2014). "CRISPR-Cas systems: beyond adaptive immunity".
*Nature Reviews Microbiology*.**12**(5): 317–326.[doi](/wiki/Doi_(identifier)):[10.1038/nrmicro3241](https://doi.org/10.1038%2Fnrmicro3241).[PMID](/wiki/PMID_(identifier))[24704746](https://pubmed.ncbi.nlm.nih.gov/24704746). - Andersson AF, Banfield JF (May 2008). "Virus population dynamics and acquired virus resistance in natural microbial communities".
*Science*.**320**(5879): 1047–1050.[Bibcode](/wiki/Bibcode_(identifier)):[2008Sci...320.1047A](https://ui.adsabs.harvard.edu/abs/2008Sci...320.1047A).[doi](/wiki/Doi_(identifier)):[10.1126/science.1157358](https://doi.org/10.1126%2Fscience.1157358).[PMID](/wiki/PMID_(identifier))[18497291](https://pubmed.ncbi.nlm.nih.gov/18497291). - Hale C, Kleppe K, Terns RM, Terns MP (December 2008).
["Prokaryotic silencing (psi)RNAs in Pyrococcus furiosus"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2590957).*RNA*.**14**(12): 2572–2579.[doi](/wiki/Doi_(identifier)):[10.1261/rna.1246808](https://doi.org/10.1261%2Frna.1246808).[PMC](/wiki/PMC_(identifier))[2590957](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2590957).[PMID](/wiki/PMID_(identifier))[18971321](https://pubmed.ncbi.nlm.nih.gov/18971321). - van der Ploeg JR (June 2009).
["Analysis of CRISPR in Streptococcus mutans suggests frequent occurrence of acquired immunity against infection by M102-like bacteriophages"](https://doi.org/10.1099%2Fmic.0.027508-0).*Microbiology*.**155**(Pt 6): 1966–1976.[doi](/wiki/Doi_(identifier)):[10.1099/mic.0.027508-0](https://doi.org/10.1099%2Fmic.0.027508-0).[PMID](/wiki/PMID_(identifier))[19383692](https://pubmed.ncbi.nlm.nih.gov/19383692). - van der Oost J, Brouns SJ (November 2009).
["RNAi: prokaryotes get in on the act"](https://doi.org/10.1016%2Fj.cell.2009.11.018).*Cell*.**139**(5): 863–865.[doi](/wiki/Doi_(identifier)):[10.1016/j.cell.2009.11.018](https://doi.org/10.1016%2Fj.cell.2009.11.018).[PMID](/wiki/PMID_(identifier))[19945373](https://pubmed.ncbi.nlm.nih.gov/19945373). - Karginov FV, Hannon GJ (January 2010).
["The CRISPR system: small RNA-guided defense in bacteria and archaea"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2819186).*Molecular Cell*.**37**(1): 7–19.[doi](/wiki/Doi_(identifier)):[10.1016/j.molcel.2009.12.033](https://doi.org/10.1016%2Fj.molcel.2009.12.033).[PMC](/wiki/PMC_(identifier))[2819186](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2819186).[PMID](/wiki/PMID_(identifier))[20129051](https://pubmed.ncbi.nlm.nih.gov/20129051). - Pul U, Wurm R, Arslan Z, Geissen R, Hofmann N, Wagner R (March 2010).
["Identification and characterization of E. coli CRISPR-cas promoters and their silencing by H-NS"](https://doi.org/10.1111%2Fj.1365-2958.2010.07073.x).*Molecular Microbiology*.**75**(6): 1495–1512.[doi](/wiki/Doi_(identifier)):[10.1111/j.1365-2958.2010.07073.x](https://doi.org/10.1111%2Fj.1365-2958.2010.07073.x).[PMID](/wiki/PMID_(identifier))[20132443](https://pubmed.ncbi.nlm.nih.gov/20132443). - Díez-Villaseñor C, Almendros C, García-Martínez J, Mojica FJ (May 2010).
["Diversity of CRISPR loci in](https://doi.org/10.1099%2Fmic.0.036046-0).*Escherichia coli*"*Microbiology*.**156**(Pt 5): 1351–1361.[doi](/wiki/Doi_(identifier)):[10.1099/mic.0.036046-0](https://doi.org/10.1099%2Fmic.0.036046-0).[PMID](/wiki/PMID_(identifier))[20133361](https://pubmed.ncbi.nlm.nih.gov/20133361). - Deveau H, Garneau JE, Moineau S (2010). "CRISPR/Cas system and its role in phage-bacteria interactions".
*Annual Review of Microbiology*.**64**: 475–493.[doi](/wiki/Doi_(identifier)):[10.1146/annurev.micro.112408.134123](https://doi.org/10.1146%2Fannurev.micro.112408.134123).[PMID](/wiki/PMID_(identifier))[20528693](https://pubmed.ncbi.nlm.nih.gov/20528693). [Koonin EV](/wiki/Eugene_Koonin), Makarova KS (December 2009).["CRISPR-Cas: an adaptive immunity system in prokaryotes"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2884157).*F1000 Biology Reports*.**1**: 95.[doi](/wiki/Doi_(identifier)):[10.3410/B1-95](https://doi.org/10.3410%2FB1-95).[PMC](/wiki/PMC_(identifier))[2884157](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2884157).[PMID](/wiki/PMID_(identifier))[20556198](https://pubmed.ncbi.nlm.nih.gov/20556198).["The age of the red pen"](https://www.economist.com/news/briefing/21661799-it-now-easy-edit-genomes-plants-animals-and-humans-age-red-pen).*The Economist*. August 22, 2015.[ISSN](/wiki/ISSN_(identifier))[0013-0613](https://search.worldcat.org/issn/0013-0613).[Archived](https://web.archive.org/web/20150824054220/http://www.economist.com/news/briefing/21661799-it-now-easy-edit-genomes-plants-animals-and-humans-age-red-pen)from the original on 2015-08-24. Retrieved 2015-08-25.- Ran AF, Hsu PD, Wright J, Agarwala V, Scott DA, Zhang F (2013).
["Genome engineering using the CRISPR-Cas9 system"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3969860).*Nature Protocols*.**8**(11): 2281–2308.[doi](/wiki/Doi_(identifier)):[10.1038/nprot.2013.143](https://doi.org/10.1038%2Fnprot.2013.143).[PMC](/wiki/PMC_(identifier))[3969860](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3969860).[PMID](/wiki/PMID_(identifier))[24157548](https://pubmed.ncbi.nlm.nih.gov/24157548).

## External links

[[edit](/w/index.php?title=CRISPR&action=edit§ion=27)]

[CRISPR](https://commons.wikimedia.org/wiki/Category:CRISPR).

["Advanced Gene Editing: CRISPR-Cas9"](https://fas.org/sgp/crs/misc/R44824.pdf)(PDF).[Congressional Research Service](/wiki/Congressional_Research_Service).["Jennifer Doudna talk: Genome Engineering with CRISPR-Cas9: Birth of a Breakthrough Technology"](https://web.archive.org/web/20161125044709/https://www.ibiology.org/ibiomagazine/jennifer-doudna-genome-engineering-with-crispr-cas9-birth-of-a-breakthrough-technology.html). 10 September 2022. Archived from[the original](https://www.ibiology.org/ibiomagazine/jennifer-doudna-genome-engineering-with-crispr-cas9-birth-of-a-breakthrough-technology.html)on 25 November 2016. Retrieved 24 November 2016.["Human Nature"](https://www.pbs.org/wgbh/nova/video/human-nature/).. Season 47. Episode 9. September 9, 2020.*NOVA*[PBS](/wiki/PBS).[WGBH](/wiki/WGBH-TV). Retrieved April 7, 2023.

### Protein Data Bank

[[edit](/w/index.php?title=CRISPR&action=edit§ion=28)]

- Overview of all the structural information available in the
[PDB](/wiki/Protein_Data_Bank)for[UniProt](/wiki/UniProt):(CRISPR system Cascade subunit CasA) at the[Q46901](https://www.ebi.ac.uk/pdbe/pdbe-kb/proteins/Q46901)[PDBe-KB](/wiki/PDBe-KB). - Overview of all the structural information available in the
[PDB](/wiki/Protein_Data_Bank)for[UniProt](/wiki/UniProt):(CRISPR system Cascade subunit CasB) at the[P76632](https://www.ebi.ac.uk/pdbe/pdbe-kb/proteins/P76632)[PDBe-KB](/wiki/PDBe-KB). - Overview of all the structural information available in the
[PDB](/wiki/Protein_Data_Bank)for[UniProt](/wiki/UniProt):(CRISPR system Cascade subunit CasC) at the[Q46899](https://www.ebi.ac.uk/pdbe/pdbe-kb/proteins/Q46899)[PDBe-KB](/wiki/PDBe-KB). - Overview of all the structural information available in the
[PDB](/wiki/Protein_Data_Bank)for[UniProt](/wiki/UniProt):(CRISPR system Cascade subunit CasD) at the[Q46898](https://www.ebi.ac.uk/pdbe/pdbe-kb/proteins/Q46898)[PDBe-KB](/wiki/PDBe-KB). - Overview of all the structural information available in the
[PDB](/wiki/Protein_Data_Bank)for[UniProt](/wiki/UniProt):(CRISPR system Cascade subunit CasE) at the[Q46897](https://www.ebi.ac.uk/pdbe/pdbe-kb/proteins/Q46897)[PDBe-KB](/wiki/PDBe-KB).